// @mastra/server/dist/server/auth/index.cjs
'use strict';

var chunkOMNWTK4B_cjs = require('../../chunk-OMNWTK4B.cjs');



Object.defineProperty(exports, "canAccessPublicly", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.canAccessPublicly; }
});
Object.defineProperty(exports, "checkRules", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.checkRules; }
});
Object.defineProperty(exports, "coreAuthMiddleware", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.coreAuthMiddleware; }
});
Object.defineProperty(exports, "defaultAuthConfig", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.defaultAuthConfig; }
});
Object.defineProperty(exports, "getAuthenticatedUser", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.getAuthenticatedUser; }
});
Object.defineProperty(exports, "isCustomRoutePublic", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.isCustomRoutePublic; }
});
Object.defineProperty(exports, "isDevPlaygroundRequest", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.isDevPlaygroundRequest; }
});
Object.defineProperty(exports, "isProtectedCustomRoute", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.isProtectedCustomRoute; }
});
Object.defineProperty(exports, "isProtectedPath", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.isProtectedPath; }
});
Object.defineProperty(exports, "matchesOrIncludes", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.matchesOrIncludes; }
});
Object.defineProperty(exports, "pathMatchesPattern", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.pathMatchesPattern; }
});
Object.defineProperty(exports, "pathMatchesRule", {
  enumerable: true,
  get: function () { return chunkOMNWTK4B_cjs.pathMatchesRule; }
});
