// @mastra/server/dist/server/handlers/a2a.cjs
'use strict';

var chunkZWKZK4WO_cjs = require('../../chunk-ZWKZK4WO.cjs');



Object.defineProperty(exports, "AGENT_EXECUTION_ROUTE", {
  enumerable: true,
  get: function () { return chunkZWKZK4WO_cjs.AGENT_EXECUTION_ROUTE; }
});
Object.defineProperty(exports, "GET_AGENT_CARD_ROUTE", {
  enumerable: true,
  get: function () { return chunkZWKZK4WO_cjs.GET_AGENT_CARD_ROUTE; }
});
Object.defineProperty(exports, "getAgentCardByIdHandler", {
  enumerable: true,
  get: function () { return chunkZWKZK4WO_cjs.getAgentCardByIdHandler; }
});
Object.defineProperty(exports, "getAgentExecutionHandler", {
  enumerable: true,
  get: function () { return chunkZWKZK4WO_cjs.getAgentExecutionHandler; }
});
Object.defineProperty(exports, "handleMessageSend", {
  enumerable: true,
  get: function () { return chunkZWKZK4WO_cjs.handleMessageSend; }
});
Object.defineProperty(exports, "handleMessageStream", {
  enumerable: true,
  get: function () { return chunkZWKZK4WO_cjs.handleMessageStream; }
});
Object.defineProperty(exports, "handleTaskCancel", {
  enumerable: true,
  get: function () { return chunkZWKZK4WO_cjs.handleTaskCancel; }
});
Object.defineProperty(exports, "handleTaskGet", {
  enumerable: true,
  get: function () { return chunkZWKZK4WO_cjs.handleTaskGet; }
});
