// @mastra/server/dist/server/handlers/prompt-block-versions.cjs
'use strict';

var chunkTEOEOSGZ_cjs = require('../../chunk-TEOEOSGZ.cjs');



Object.defineProperty(exports, "ACTIVATE_PROMPT_BLOCK_VERSION_ROUTE", {
  enumerable: true,
  get: function () { return chunkTEOEOSGZ_cjs.ACTIVATE_PROMPT_BLOCK_VERSION_ROUTE; }
});
Object.defineProperty(exports, "COMPARE_PROMPT_BLOCK_VERSIONS_ROUTE", {
  enumerable: true,
  get: function () { return chunkTEOEOSGZ_cjs.COMPARE_PROMPT_BLOCK_VERSIONS_ROUTE; }
});
Object.defineProperty(exports, "CREATE_PROMPT_BLOCK_VERSION_ROUTE", {
  enumerable: true,
  get: function () { return chunkTEOEOSGZ_cjs.CREATE_PROMPT_BLOCK_VERSION_ROUTE; }
});
Object.defineProperty(exports, "DELETE_PROMPT_BLOCK_VERSION_ROUTE", {
  enumerable: true,
  get: function () { return chunkTEOEOSGZ_cjs.DELETE_PROMPT_BLOCK_VERSION_ROUTE; }
});
Object.defineProperty(exports, "GET_PROMPT_BLOCK_VERSION_ROUTE", {
  enumerable: true,
  get: function () { return chunkTEOEOSGZ_cjs.GET_PROMPT_BLOCK_VERSION_ROUTE; }
});
Object.defineProperty(exports, "LIST_PROMPT_BLOCK_VERSIONS_ROUTE", {
  enumerable: true,
  get: function () { return chunkTEOEOSGZ_cjs.LIST_PROMPT_BLOCK_VERSIONS_ROUTE; }
});
Object.defineProperty(exports, "RESTORE_PROMPT_BLOCK_VERSION_ROUTE", {
  enumerable: true,
  get: function () { return chunkTEOEOSGZ_cjs.RESTORE_PROMPT_BLOCK_VERSION_ROUTE; }
});
