// @mastra/server/dist/server/handlers/stored-skills.cjs
'use strict';

var chunkE5JCKPP3_cjs = require('../../chunk-E5JCKPP3.cjs');



Object.defineProperty(exports, "CREATE_STORED_SKILL_ROUTE", {
  enumerable: true,
  get: function () { return chunkE5JCKPP3_cjs.CREATE_STORED_SKILL_ROUTE; }
});
Object.defineProperty(exports, "DELETE_STORED_SKILL_ROUTE", {
  enumerable: true,
  get: function () { return chunkE5JCKPP3_cjs.DELETE_STORED_SKILL_ROUTE; }
});
Object.defineProperty(exports, "GET_STORED_SKILL_ROUTE", {
  enumerable: true,
  get: function () { return chunkE5JCKPP3_cjs.GET_STORED_SKILL_ROUTE; }
});
Object.defineProperty(exports, "LIST_STORED_SKILLS_ROUTE", {
  enumerable: true,
  get: function () { return chunkE5JCKPP3_cjs.LIST_STORED_SKILLS_ROUTE; }
});
Object.defineProperty(exports, "PUBLISH_STORED_SKILL_ROUTE", {
  enumerable: true,
  get: function () { return chunkE5JCKPP3_cjs.PUBLISH_STORED_SKILL_ROUTE; }
});
Object.defineProperty(exports, "UPDATE_STORED_SKILL_ROUTE", {
  enumerable: true,
  get: function () { return chunkE5JCKPP3_cjs.UPDATE_STORED_SKILL_ROUTE; }
});
