// @mastra/server/dist/server/handlers/stored-agents.cjs
'use strict';

var chunkA4UUHYOD_cjs = require('../../chunk-A4UUHYOD.cjs');



Object.defineProperty(exports, "CREATE_STORED_AGENT_ROUTE", {
  enumerable: true,
  get: function () { return chunkA4UUHYOD_cjs.CREATE_STORED_AGENT_ROUTE; }
});
Object.defineProperty(exports, "DELETE_STORED_AGENT_ROUTE", {
  enumerable: true,
  get: function () { return chunkA4UUHYOD_cjs.DELETE_STORED_AGENT_ROUTE; }
});
Object.defineProperty(exports, "GET_STORED_AGENT_ROUTE", {
  enumerable: true,
  get: function () { return chunkA4UUHYOD_cjs.GET_STORED_AGENT_ROUTE; }
});
Object.defineProperty(exports, "LIST_STORED_AGENTS_ROUTE", {
  enumerable: true,
  get: function () { return chunkA4UUHYOD_cjs.LIST_STORED_AGENTS_ROUTE; }
});
Object.defineProperty(exports, "PREVIEW_INSTRUCTIONS_ROUTE", {
  enumerable: true,
  get: function () { return chunkA4UUHYOD_cjs.PREVIEW_INSTRUCTIONS_ROUTE; }
});
Object.defineProperty(exports, "UPDATE_STORED_AGENT_ROUTE", {
  enumerable: true,
  get: function () { return chunkA4UUHYOD_cjs.UPDATE_STORED_AGENT_ROUTE; }
});
