// @mastra/server/dist/server/handlers/vector.cjs
'use strict';

var chunkJN4QR7QO_cjs = require('../../chunk-JN4QR7QO.cjs');



Object.defineProperty(exports, "CREATE_INDEX_ROUTE", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.CREATE_INDEX_ROUTE; }
});
Object.defineProperty(exports, "DELETE_INDEX_ROUTE", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.DELETE_INDEX_ROUTE; }
});
Object.defineProperty(exports, "DESCRIBE_INDEX_ROUTE", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.DESCRIBE_INDEX_ROUTE; }
});
Object.defineProperty(exports, "LIST_EMBEDDERS_ROUTE", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.LIST_EMBEDDERS_ROUTE; }
});
Object.defineProperty(exports, "LIST_INDEXES_ROUTE", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.LIST_INDEXES_ROUTE; }
});
Object.defineProperty(exports, "LIST_VECTORS_ROUTE", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.LIST_VECTORS_ROUTE; }
});
Object.defineProperty(exports, "QUERY_VECTORS_ROUTE", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.QUERY_VECTORS_ROUTE; }
});
Object.defineProperty(exports, "UPSERT_VECTORS_ROUTE", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.UPSERT_VECTORS_ROUTE; }
});
Object.defineProperty(exports, "createIndex", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.createIndex; }
});
Object.defineProperty(exports, "deleteIndex", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.deleteIndex; }
});
Object.defineProperty(exports, "describeIndex", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.describeIndex; }
});
Object.defineProperty(exports, "listIndexes", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.listIndexes; }
});
Object.defineProperty(exports, "listVectorStores", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.listVectorStores; }
});
Object.defineProperty(exports, "queryVectors", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.queryVectors; }
});
Object.defineProperty(exports, "upsertVectors", {
  enumerable: true,
  get: function () { return chunkJN4QR7QO_cjs.upsertVectors; }
});
