// @mastra/server/dist/server/handlers/voice.cjs
'use strict';

var chunkF4TKO2VD_cjs = require('../../chunk-F4TKO2VD.cjs');



Object.defineProperty(exports, "GENERATE_SPEECH_DEPRECATED_ROUTE", {
  enumerable: true,
  get: function () { return chunkF4TKO2VD_cjs.GENERATE_SPEECH_DEPRECATED_ROUTE; }
});
Object.defineProperty(exports, "GENERATE_SPEECH_ROUTE", {
  enumerable: true,
  get: function () { return chunkF4TKO2VD_cjs.GENERATE_SPEECH_ROUTE; }
});
Object.defineProperty(exports, "GET_LISTENER_ROUTE", {
  enumerable: true,
  get: function () { return chunkF4TKO2VD_cjs.GET_LISTENER_ROUTE; }
});
Object.defineProperty(exports, "GET_SPEAKERS_DEPRECATED_ROUTE", {
  enumerable: true,
  get: function () { return chunkF4TKO2VD_cjs.GET_SPEAKERS_DEPRECATED_ROUTE; }
});
Object.defineProperty(exports, "GET_SPEAKERS_ROUTE", {
  enumerable: true,
  get: function () { return chunkF4TKO2VD_cjs.GET_SPEAKERS_ROUTE; }
});
Object.defineProperty(exports, "TRANSCRIBE_SPEECH_DEPRECATED_ROUTE", {
  enumerable: true,
  get: function () { return chunkF4TKO2VD_cjs.TRANSCRIBE_SPEECH_DEPRECATED_ROUTE; }
});
Object.defineProperty(exports, "TRANSCRIBE_SPEECH_ROUTE", {
  enumerable: true,
  get: function () { return chunkF4TKO2VD_cjs.TRANSCRIBE_SPEECH_ROUTE; }
});
