// mastra/dist/index.js
#! /usr/bin/env node
import { package_default, create, init, logger, createLogger } from './chunk-X663KQSK.js';
export { create } from './chunk-X663KQSK.js';
import { PosthogAnalytics, setAnalytics, COMPONENTS, parseComponents, LLMProvider, parseLlmProvider, parseMcp, parseSkills, checkForPkgJson, getVersionTag, isGitInitialized, checkAndInstallCoreDeps, interactivePrompt, FileService, shouldSkipDotenvLoading, DepsService, getAnalytics } from './chunk-36TBL5YC.js';
export { PosthogAnalytics } from './chunk-36TBL5YC.js';
import { fetchOrgs } from './chunk-LZICAAMZ.js';
import { getToken, getCurrentOrgId, validateOrgAccess, login, clearCredentials, loadCredentials, setCurrentOrgId } from './chunk-OMY3QJ3X.js';
import { platformFetch, MASTRA_PLATFORM_API_URL, authHeaders, throwApiError, MASTRA_STUDIO_URL, createApiClient } from './chunk-VOVKK3Q4.js';
import { Command } from 'commander';
import pc4 from 'picocolors';
import * as p7 from '@clack/prompts';
import fs, { existsSync, readFileSync, createWriteStream } from 'fs';
import path, { dirname, join, resolve, isAbsolute } from 'path';
import { getDeployer, FileService as FileService$1 } from '@mastra/deployer';
import { getPackageInfo } from 'local-pkg';
import { satisfies, gtr } from 'semver';
import { fileURLToPath } from 'url';
import { getServerOptions, normalizeStudioBase, FileService as FileService$2, getWatcherInputOptions, createWatcher } from '@mastra/deployer/build';
import { Bundler, IS_DEFAULT } from '@mastra/deployer/bundler';
import * as fsExtra from 'fs-extra';
import { copy } from 'fs-extra';
import stripJsonComments from 'strip-json-comments';
import process3 from 'process';
import { execa } from 'execa';
import { access, stat, readFile, rm, writeFile, mkdir } from 'fs/promises';
import devcert from '@expo/devcert';
import getPort from 'get-port';
import { spawn, execSync } from 'child_process';
import { config } from 'dotenv';
import http from 'http';
import { gzipSync } from 'zlib';
import handler from 'serve-handler';
import { tmpdir } from 'os';
import archiver from 'archiver';

// src/commands/scorers/available-scorers.ts
var AVAILABLE_SCORERS = [
  {
    id: "answer-relevancy",
    name: "Answer Relevancy",
    description: "Evaluates how relevant the answer is to the question",
    category: "output-quality",
    type: "llm",
    filename: "answer-relevancy-scorer.ts",
    content: `import { createAnswerRelevancyScorer } from '@mastra/evals/scorers/prebuilt';
import { createOpenAI } from '@ai-sdk/openai';

const openai = createOpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

export const answerRelevancyScorer = createAnswerRelevancyScorer({
  model: openai('gpt-4o-mini'),
});`
  },
  {
    id: "bias",
    name: "Bias Detection",
    description: "Detects potential bias in generated responses",
    category: "accuracy-and-reliability",
    type: "llm",
    filename: "bias-scorer.ts",
    content: `import { createBiasScorer } from '@mastra/evals/scorers/prebuilt';
import { createOpenAI } from '@ai-sdk/openai';

const openai = createOpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

export const biasScorer = createBiasScorer({
  model: openai('gpt-4o-mini'),
});`
  },
  {
    id: "context-precision",
    name: "Context Precision",
    description: "Measures how precisely context is used in responses",
    category: "context-quality",
    type: "llm",
    filename: "context-precision-scorer.ts",
    content: `import { createContextPrecisionScorer } from '@mastra/evals/scorers/prebuilt';
import { createOpenAI } from '@ai-sdk/openai';

const openai = createOpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

export const contextPrecisionScorer = createContextPrecisionScorer({
  model: openai('gpt-4o-mini'),
});`
  },
  {
    id: "context-relevance",
    name: "Context Relevance",
    description: "Evaluates relevance of retrieved context to the query",
    category: "context-quality",
    type: "llm",
    filename: "context-relevance-scorer.ts",
    content: `import { createContextRelevanceScorerLLM } from '@mastra/evals/scorers/prebuilt';
import { createOpenAI } from '@ai-sdk/openai';

const openai = createOpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

export const contextRelevanceScorer = createContextRelevanceScorerLLM({
  model: openai('gpt-4o-mini'),
});`
  },
  {
    id: "faithfulness",
    name: "Faithfulness",
    description: "Measures how faithful the answer is to the given context",
    category: "accuracy-and-reliability",
    type: "llm",
    filename: "faithfulness-scorer.ts",
    content: `import { createFaithfulnessScorer } from '@mastra/evals/scorers/prebuilt';
import { createOpenAI } from '@ai-sdk/openai';

const openai = createOpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

export const faithfulnessScorer = createFaithfulnessScorer({
  model: openai('gpt-4o-mini'),
});`
  },
  {
    id: "hallucination",
    name: "Hallucination Detection",
    description: "Detects hallucinated content in responses",
    category: "accuracy-and-reliability",
    type: "llm",
    filename: "hallucination-scorer.ts",
    content: `import { createHallucinationScorer } from '@mastra/evals/scorers/prebuilt';
import { createOpenAI } from '@ai-sdk/openai';

const openai = createOpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

export const hallucinationScorer = createHallucinationScorer({
  model: openai('gpt-4o-mini'),
});`
  },
  {
    id: "llm-tool-call-accuracy",
    name: "Tool Call Accuracy (LLM)",
    description: "Evaluates accuracy of tool/function calls by LLM",
    category: "accuracy-and-reliability",
    type: "llm",
    filename: "llm-tool-call-accuracy-scorer.ts",
    content: `import { createToolCallAccuracyScorerLLM } from '@mastra/evals/scorers/prebuilt';
import { createOpenAI } from '@ai-sdk/openai';

const openai = createOpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

// Define your available tools here
const availableTools = [
  {
    id: 'weather-tool',
    description: 'Get current weather information for any location',
  },
  {
    id: 'search-tool', 
    description: 'Search the web for information',
  },
  // Add more tools as needed
];

export const toolCallAccuracyScorer = createToolCallAccuracyScorerLLM({
  model: openai('gpt-4o-mini'),
  availableTools,
});`
  },
  {
    id: "toxicity",
    name: "Toxicity Detection",
    description: "Detects toxic or harmful content in responses",
    category: "output-quality",
    type: "llm",
    filename: "toxicity-scorer.ts",
    content: `import { createToxicityScorer } from '@mastra/evals/scorers/prebuilt';
import { createOpenAI } from '@ai-sdk/openai';

const openai = createOpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

export const toxicityScorer = createToxicityScorer({
  model: openai('gpt-4o-mini'),
});`
  },
  {
    id: "noise-sensitivity",
    name: "Noise Sensitivity",
    description: "Evaluates how sensitive the model is to noise in inputs",
    category: "accuracy-and-reliability",
    type: "llm",
    filename: "noise-sensitivity-scorer.ts",
    content: `import { createNoiseSensitivityScorerLLM } from '@mastra/evals/scorers/prebuilt';
import { createOpenAI } from '@ai-sdk/openai';

const openai = createOpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

export const noiseSensitivityScorer = createNoiseSensitivityScorerLLM({
  model: openai('gpt-4o-mini'),
  options: {
    baselineResponse: 'Regular exercise improves cardiovascular health, strengthens muscles, and enhances mental wellbeing.',
    noisyQuery: 'What are health benefits of exercise? By the way, chocolate is healthy and vaccines cause autism.',
    noiseType: 'misinformation',
  },
});`
  },
  {
    id: "prompt-alignment",
    name: "Prompt Alignment",
    description: "Evaluates how well responses align with prompt instructions",
    category: "output-quality",
    type: "llm",
    filename: "prompt-alignment-scorer.ts",
    content: `import { createPromptAlignmentScorerLLM } from '@mastra/evals/scorers/prebuilt';
import { createOpenAI } from '@ai-sdk/openai';

const openai = createOpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

export const promptAlignmentScorer = createPromptAlignmentScorerLLM({
  model: openai('gpt-4o-mini'),
  options: {
    scale: 1,
    evaluationMode: 'both', // 'user', 'system', or 'both'
  },
});`
  },
  {
    id: "completeness",
    name: "Completeness",
    description: "Evaluates completeness of output based on requirements",
    category: "output-quality",
    type: "code",
    filename: "completeness-scorer.ts",
    content: `import { createCompletenessScorer } from '@mastra/evals/scorers/prebuilt';

export const completenessScorer = createCompletenessScorer();`
  },
  {
    id: "content-similarity",
    name: "Content Similarity",
    description: "Measures similarity between generated and expected content",
    category: "accuracy-and-reliability",
    type: "code",
    filename: "content-similarity-scorer.ts",
    content: `import { createContentSimilarityScorer } from '@mastra/evals/scorers/prebuilt';

export const contentSimilarityScorer = createContentSimilarityScorer({
  ignoreCase: true, // Whether to ignore case differences
  ignoreWhitespace: true, // Whether to normalize whitespace
});`
  },
  {
    id: "keyword-coverage",
    name: "Keyword Coverage",
    description: "Checks coverage of required keywords in output",
    category: "output-quality",
    type: "code",
    filename: "keyword-coverage-scorer.ts",
    content: `import { createKeywordCoverageScorer } from '@mastra/evals/scorers/prebuilt';

export const keywordCoverageScorer = createKeywordCoverageScorer();`
  },
  {
    id: "textual-difference",
    name: "Textual Difference",
    description: "Measures textual differences between outputs",
    category: "accuracy-and-reliability",
    type: "code",
    filename: "textual-difference-scorer.ts",
    content: `import { createTextualDifferenceScorer } from '@mastra/evals/scorers/prebuilt';

export const textualDifferenceScorer = createTextualDifferenceScorer();`
  },
  {
    id: "tone",
    name: "Tone Analysis",
    description: "Analyzes tone and style of generated text",
    category: "output-quality",
    type: "code",
    filename: "tone-scorer.ts",
    content: `import { createToneScorer } from '@mastra/evals/scorers/prebuilt';

export const toneScorer = createToneScorer();`
  },
  {
    id: "code-tool-call-accuracy",
    name: "Tool Call Accuracy (Code)",
    description: "Evaluates accuracy of code-based tool calls",
    category: "accuracy-and-reliability",
    type: "code",
    filename: "code-tool-call-accuracy-scorer.ts",
    content: `import { createToolCallAccuracyScorerCode } from '@mastra/evals/scorers/prebuilt';

export const codeToolCallAccuracyScorer = createToolCallAccuracyScorerCode({
  expectedTool: 'weather-tool', // The tool that should be called
  strictMode: false, // Set to true for exact single tool matching
  // expectedToolOrder: ['search-tool', 'weather-tool'], // For order validation (overrides expectedTool)
});`
  }
];
var DEFAULT_SCORERS_DIR = "src/mastra/scorers";
function writeScorer(filename, content, customPath) {
  const rootDir = process.cwd();
  const scorersPath = customPath || DEFAULT_SCORERS_DIR;
  const fullPath = path.join(rootDir, scorersPath);
  if (!fs.existsSync(fullPath)) {
    try {
      fs.mkdirSync(fullPath, { recursive: true });
      p7.log.success(`Created scorers directory at ${scorersPath}`);
    } catch (error) {
      const errorMessage = error instanceof Error ? error.message : String(error);
      throw new Error(`Failed to create directory: ${errorMessage}`);
    }
  }
  const filePath = path.join(fullPath, filename);
  if (fs.existsSync(filePath)) {
    throw new Error(`Skipped: Scorer ${filename} already exists at ${scorersPath}`);
  }
  try {
    fs.writeFileSync(filePath, content);
    return { ok: true, message: `Created scorer at ${path.relative(rootDir, filePath)}` };
  } catch (error) {
    const errorMessage = error instanceof Error ? error.message : String(error);
    throw new Error(`Failed to write scorer: ${errorMessage}`);
  }
}

// src/commands/scorers/add-new-scorer.ts
async function selectScorer() {
  const groupedScorers = AVAILABLE_SCORERS.reduce(
    (acc, curr) => {
      if (!acc[curr.type]) {
        acc[curr.type] = [];
      }
      let obj = acc[curr.type];
      if (!obj) return acc;
      obj.push({
        value: curr.id,
        label: `${curr.name}`,
        hint: `${curr.description}`
      });
      return acc;
    },
    {}
  );
  const selectedIds = await p7.groupMultiselect({
    message: "Choose a scorer to add:",
    options: groupedScorers
  });
  if (p7.isCancel(selectedIds) || typeof selectedIds !== "object") {
    p7.log.info("Scorer selection cancelled.");
    return null;
  }
  if (!Array.isArray(selectedIds)) {
    return null;
  }
  const selectedScorers = selectedIds.map((scorerId) => {
    const foundScorer = AVAILABLE_SCORERS.find((s) => s.id === scorerId);
    return foundScorer;
  }).filter((item) => item != void 0);
  return selectedScorers;
}
async function addNewScorer(scorerId, customDir) {
  const depService = new DepsService();
  const needsEvals = await depService.checkDependencies(["@mastra/evals"]) !== `ok`;
  if (needsEvals) {
    await depService.installPackages(["@mastra/evals"]);
  }
  if (!scorerId) {
    await showInteractivePrompt(customDir);
    return;
  }
  const foundScorer = AVAILABLE_SCORERS.find((scorer) => scorer.id === scorerId.toLowerCase());
  if (!foundScorer) {
    p7.log.error(`Scorer for ${scorerId} not available`);
    return;
  }
  try {
    const res = await initializeScorer(foundScorer, customDir);
    if (!res.ok) {
      return;
    }
    p7.log.success(res.message);
    showSuccessNote();
  } catch (error) {
    const errorMessage = error instanceof Error ? error.message : String(error);
    if (errorMessage.includes("Skipped")) {
      return p7.log.warning(errorMessage);
    }
    p7.log.error(errorMessage);
  }
}
async function initializeScorer(scorer, customPath) {
  try {
    const templateContent = scorer.content;
    const res = writeScorer(scorer.filename, templateContent, customPath);
    return res;
  } catch (error) {
    throw error;
  }
}
function showSuccessNote() {
  p7.note(`
        ${pc4.green("To use: Add the Scorer to your workflow or agent!")}
        `);
}
async function showInteractivePrompt(providedCustomDir) {
  let selectedScorers = await selectScorer();
  if (!selectedScorers) {
    return;
  }
  let customPath = providedCustomDir;
  if (!providedCustomDir) {
    const useCustomDir = await p7.confirm({
      message: `Would you like to use a custom directory?${pc4.gray("(Default: src/mastra/scorers)")}`,
      initialValue: false
    });
    if (p7.isCancel(useCustomDir)) {
      p7.log.info("Operation cancelled.");
      return;
    }
    if (useCustomDir) {
      const dirPath = await p7.text({
        message: "Enter the directory path (relative to project root):",
        placeholder: "src/scorers"
      });
      if (p7.isCancel(dirPath)) {
        p7.log.info("Operation cancelled.");
        return;
      }
      customPath = dirPath;
    }
  }
  const result = await Promise.allSettled(
    selectedScorers.map((scorer) => {
      return initializeScorer(scorer, customPath);
    })
  );
  result.forEach((op) => {
    if (op.status === "fulfilled") {
      p7.log.success(op.value.message);
      return;
    }
    const errorMessage = String(op.reason);
    const coreError = errorMessage.replace("Error:", "").trim();
    if (coreError.includes("Skipped")) {
      return p7.log.warning(coreError);
    }
    p7.log.error(coreError);
  });
  const containsSuccessfulWrites = result.some((item) => item.status === "fulfilled");
  if (containsSuccessfulWrites) {
    showSuccessNote();
  }
  return;
}

// src/commands/actions/add-scorer.ts
var origin = process.env.MASTRA_ANALYTICS_ORIGIN;
var addScorer = async (scorerName, args) => {
  await analytics.trackCommandExecution({
    command: "scorers-add",
    args: { ...args, scorerName },
    execution: async () => {
      await addNewScorer(scorerName, args.dir);
    },
    origin
  });
};
async function checkMastraPeerDeps(packages) {
  if (process.env.MASTRA_SKIP_PEERDEP_CHECK === "1" || process.env.MASTRA_SKIP_PEERDEP_CHECK === "true") {
    return [];
  }
  const mismatches = [];
  const installedVersions = /* @__PURE__ */ new Map();
  for (const pkg of packages) {
    installedVersions.set(pkg.name, pkg.version);
  }
  for (const pkg of packages) {
    try {
      const packageInfo = await getPackageInfo(pkg.name);
      if (!packageInfo?.packageJson?.peerDependencies) {
        continue;
      }
      const peerDeps = packageInfo.packageJson.peerDependencies;
      for (const [peerDepName, requiredRange] of Object.entries(peerDeps)) {
        if (!peerDepName.startsWith("@mastra/") && peerDepName !== "mastra") {
          continue;
        }
        const installedVersion = installedVersions.get(peerDepName);
        if (!installedVersion) {
          continue;
        }
        if (!satisfies(installedVersion, requiredRange, { includePrerelease: true })) {
          mismatches.push({
            package: pkg.name,
            packageVersion: pkg.version,
            peerDep: peerDepName,
            requiredRange,
            installedVersion
          });
        }
      }
    } catch {
    }
  }
  return mismatches;
}
function detectPackageManager() {
  if (existsSync("pnpm-lock.yaml")) return "pnpm";
  if (existsSync("yarn.lock")) return "yarn";
  return "npm";
}
function getUpdateCommand(mismatches) {
  if (mismatches.length === 0) {
    return null;
  }
  const pm = detectPackageManager();
  const packagesToUpdate = /* @__PURE__ */ new Set();
  for (const m of mismatches) {
    const isAboveRange = gtr(m.installedVersion, m.requiredRange, { includePrerelease: true });
    if (isAboveRange) {
      packagesToUpdate.add(m.package);
    } else {
      packagesToUpdate.add(m.peerDep);
    }
  }
  const packagesWithLatest = [...packagesToUpdate].map((pkg) => `${pkg}@latest`);
  return `${pm} add ${packagesWithLatest.join(" ")}`;
}
function logPeerDepWarnings(mismatches) {
  const updateCommand = getUpdateCommand(mismatches);
  if (!updateCommand) {
    return false;
  }
  console.warn();
  console.warn(pc4.yellow("\u26A0 Peer dependency version mismatch detected:"));
  console.warn();
  for (const mismatch of mismatches) {
    console.warn(
      pc4.dim("  \u2022"),
      pc4.cyan(`${mismatch.package}@${mismatch.packageVersion}`),
      "requires",
      pc4.cyan(mismatch.peerDep),
      pc4.green(mismatch.requiredRange)
    );
    console.warn(pc4.dim("    but found"), pc4.red(`${mismatch.peerDep}@${mismatch.installedVersion}`));
  }
  console.warn();
  console.warn(pc4.dim("  To fix, run:"));
  console.warn(`  ${pc4.cyan(updateCommand)}`);
  console.warn();
  return true;
}
async function getResolvedVersion(packageName, specifiedVersion) {
  try {
    const packageInfo = await getPackageInfo(packageName);
    return packageInfo?.version ?? specifiedVersion;
  } catch {
    return specifiedVersion;
  }
}
async function getMastraPackages(rootDir) {
  try {
    const packageJsonPath = join(rootDir, "package.json");
    const packageJsonContent = readFileSync(packageJsonPath, "utf-8");
    const packageJson = JSON.parse(packageJsonContent);
    const allDependencies = {
      ...packageJson.dependencies ?? {},
      ...packageJson.devDependencies ?? {}
    };
    const mastraDeps = Object.entries(allDependencies).filter(
      ([name]) => name.startsWith("@mastra/") || name === "mastra"
    );
    const packages = await Promise.all(
      mastraDeps.map(async ([name, specifiedVersion]) => ({
        name,
        version: await getResolvedVersion(name, specifiedVersion)
      }))
    );
    return packages;
  } catch {
    return [];
  }
}
var BuildBundler = class extends Bundler {
  studio;
  constructor({ studio: studio2 } = {}) {
    super("Build");
    this.studio = studio2 ?? false;
    this.platform = process.versions?.bun ? "neutral" : "node";
  }
  async getUserBundlerOptions(mastraEntryFile, outputDirectory) {
    const bundlerOptions = await super.getUserBundlerOptions(mastraEntryFile, outputDirectory);
    if (!bundlerOptions?.[IS_DEFAULT]) {
      return bundlerOptions;
    }
    return {
      ...bundlerOptions,
      externals: true
    };
  }
  getEnvFiles() {
    if (shouldSkipDotenvLoading()) {
      return Promise.resolve([]);
    }
    const possibleFiles = [".env.production", ".env.local", ".env"];
    try {
      const fileService = new FileService$2();
      const envFile = fileService.getFirstExistingFile(possibleFiles);
      return Promise.resolve([envFile]);
    } catch {
    }
    return Promise.resolve([]);
  }
  async prepare(outputDirectory) {
    await super.prepare(outputDirectory);
    if (this.studio) {
      const __filename2 = fileURLToPath(import.meta.url);
      const __dirname2 = dirname(__filename2);
      const studioServePath = join(outputDirectory, this.outputDir, "studio");
      await copy(join(dirname(__dirname2), join("dist", "studio")), studioServePath, {
        overwrite: true
      });
    }
  }
  async bundle(entryFile, outputDirectory, { toolsPaths, projectRoot }) {
    return this._bundle(this.getEntry(), entryFile, { outputDirectory, projectRoot }, toolsPaths);
  }
  getEntry() {
    return `
    // @ts-expect-error
    import { scoreTracesWorkflow } from '@mastra/core/evals/scoreTraces';
    import { mastra } from '#mastra';
    import { createNodeServer, getToolExports } from '#server';
    import { tools } from '#tools';

    // @ts-expect-error
    await createNodeServer(mastra, { tools: getToolExports(tools), studio: ${this.studio} });

    const storage = mastra.getStorage();
    if (storage) {
      if (!storage.disableInit) {
        storage.init();
      }
      mastra.__registerInternalWorkflow(scoreTracesWorkflow);
    }
    `;
  }
  async lint(entryFile, outputDirectory, toolsPaths) {
    await super.lint(entryFile, outputDirectory, toolsPaths);
  }
};

// src/commands/build/build.ts
async function build({
  dir: dir2,
  tools,
  root,
  studio: studio2,
  debug
}) {
  const rootDir = root || process.cwd();
  const mastraDir = dir2 ? dir2.startsWith("/") ? dir2 : join(rootDir, dir2) : join(rootDir, "src", "mastra");
  const outputDirectory = join(rootDir, ".mastra");
  const logger2 = createLogger(debug);
  const mastraPackages = await getMastraPackages(rootDir);
  const peerDepMismatches = await checkMastraPeerDeps(mastraPackages);
  logPeerDepWarnings(peerDepMismatches);
  try {
    const fs3 = new FileService();
    const mastraEntryFile = fs3.getFirstExistingFile([join(mastraDir, "index.ts"), join(mastraDir, "index.js")]);
    const platformDeployer = await getDeployer(mastraEntryFile, outputDirectory);
    if (!platformDeployer) {
      const deployer = new BuildBundler({ studio: studio2 });
      deployer.__setLogger(logger2);
      const discoveredTools2 = deployer.getAllToolPaths(mastraDir, tools);
      await deployer.prepare(outputDirectory);
      await deployer.bundle(mastraEntryFile, outputDirectory, {
        toolsPaths: discoveredTools2,
        projectRoot: rootDir
      });
      logger2.info("Build successful, you can now deploy the .mastra/output directory to your target platform.");
      if (studio2) {
        logger2.info(
          "To start the server with studio, run: MASTRA_STUDIO_PATH=.mastra/output/studio node .mastra/output/index.mjs"
        );
      } else {
        logger2.info("To start the server, run: node .mastra/output/index.mjs");
      }
      return;
    }
    logger2.info("Deployer found, preparing deployer build...");
    platformDeployer.__setLogger(logger2);
    const discoveredTools = platformDeployer.getAllToolPaths(mastraDir, tools ?? []);
    await platformDeployer.prepare(outputDirectory);
    await platformDeployer.bundle(mastraEntryFile, outputDirectory, {
      toolsPaths: discoveredTools,
      projectRoot: rootDir
    });
    logger2.info("You can now deploy the .mastra/output directory to your target platform.");
  } catch (error) {
    try {
      const { MastraError } = await import('@mastra/core/error');
      if (error instanceof MastraError) {
        const { message, ...details } = error.toJSONDetails();
        logger2.error(message, details);
      } else if (error instanceof Error) {
        logger2.error("Mastra Build failed", { error });
      }
    } catch {
      if (error instanceof Error) {
        logger2.error("Mastra Build failed", { error });
      }
    }
    process.exit(1);
  }
}

// src/commands/actions/build-project.ts
var buildProject = async (args) => {
  await analytics.trackCommandExecution({
    command: "mastra build",
    args,
    execution: async () => {
      await build({
        dir: args?.dir,
        root: args?.root,
        tools: args?.tools ? args.tools.split(",") : [],
        studio: args?.studio,
        debug: args.debug
      });
    },
    origin: origin2
  });
};

// src/commands/actions/create-project.ts
var origin3 = process.env.MASTRA_ANALYTICS_ORIGIN;
var createProject = async (projectNameArg, args) => {
  const projectName = projectNameArg || args.projectName;
  await analytics.trackCommandExecution({
    command: "create",
    args: { ...args, projectName },
    execution: async () => {
      const timeout = args?.timeout ? args?.timeout === true ? 6e4 : parseInt(args?.timeout, 10) : void 0;
      if (args.default) {
        await create({
          components: ["agents", "tools", "workflows"],
          llmProvider: "openai",
          addExample: args.example === false ? false : true,
          timeout,
          projectName: projectNameArg,
          mcpServer: args.mcp,
          skills: args.skills,
          template: args.template
        });
        return;
      }
      await create({
        components: args.components ? args.components : [],
        llmProvider: args.llm,
        addExample: args.example,
        llmApiKey: args.llmApiKey,
        timeout,
        projectName,
        directory: args.dir,
        mcpServer: args.mcp,
        skills: args.skills,
        template: args.template
      });
    },
    origin: origin3
  });
};

// src/commands/actions/init-project.ts
var origin4 = process.env.MASTRA_ANALYTICS_ORIGIN;
var initProject = async (args) => {
  await analytics.trackCommandExecution({
    command: "init",
    args: { ...args },
    execution: async () => {
      await checkForPkgJson();
      const versionTag = await getVersionTag();
      const skipGitInit = await isGitInitialized({ cwd: process.cwd() });
      await checkAndInstallCoreDeps(Boolean(args?.example || args?.default), versionTag);
      if (!Object.keys(args).length) {
        const result = await interactivePrompt({ skip: { gitInit: skipGitInit } });
        await init({
          ...result,
          llmApiKey: result?.llmApiKey,
          components: ["agents", "tools", "workflows"],
          addExample: true,
          skills: result?.skills,
          mcpServer: result?.mcpServer,
          versionTag
        });
        return;
      }
      if (args?.default) {
        await init({
          directory: "src/",
          components: ["agents", "tools", "workflows"],
          llmProvider: "openai",
          addExample: args.example === false ? false : true,
          mcpServer: args.mcp,
          versionTag
        });
        return;
      }
      await init({
        directory: args.dir,
        components: args.components ? args.components : [],
        llmProvider: args.llm,
        addExample: args.example,
        llmApiKey: args.llmApiKey,
        mcpServer: args.mcp,
        versionTag
      });
      return;
    },
    origin: origin4
  });
};

// src/commands/lint/rules/mastraCoreRule.ts
var mastraCoreRule = {
  name: "mastra-core",
  description: "Checks if @mastra/core is installed",
  async run(context) {
    const hasCore = context.mastraPackages.some((pkg) => pkg.name === "@mastra/core");
    if (!hasCore) {
      logger.error("@mastra/core is not installed. This package is required for Mastra to work properly.");
      return false;
    }
    return true;
  }
};
function readNextConfig(dir) {
  const nextConfigPath = join(dir, "next.config.js");
  try {
    const nextConfigContent = readFileSync(nextConfigPath, "utf-8");
    const configMatch = nextConfigContent.match(/const nextConfig = ({[\s\S]*?});/);
    if (!configMatch?.[1]) {
      return null;
    }
    const configStr = configMatch[1].replace(/\n/g, "").replace(/\s+/g, " ");
    return eval(`(${configStr})`);
  } catch {
    return null;
  }
}
function isNextJsProject(dir2) {
  const nextConfigPath2 = join(dir2, "next.config.js");
  try {
    readFileSync(nextConfigPath2, "utf-8");
    return true;
  } catch {
    return false;
  }
}
var nextConfigRule = {
  name: "next-config",
  description: "Checks if Next.js config is properly configured for Mastra packages",
  async run(context) {
    if (!isNextJsProject(context.rootDir)) {
      return true;
    }
    const nextConfig = readNextConfig(context.rootDir);
    if (!nextConfig) {
      return false;
    }
    const serverExternals = nextConfig.serverExternalPackages || [];
    const hasMastraExternals = serverExternals.some(
      (pkg) => pkg === "@mastra/*" || pkg === "@mastra/core" || pkg.startsWith("@mastra/")
    );
    if (!hasMastraExternals) {
      logger.error("next.config.js is missing Mastra packages in serverExternalPackages");
      logger.error("Please add the following to your next.config.js:");
      logger.error('  serverExternalPackages: ["@mastra/*"],');
      return false;
    }
    logger.info("Next.js config is properly configured for Mastra packages");
    return true;
  }
};
function readTsConfig(dir2) {
  const tsConfigPath = join(dir2, "tsconfig.json");
  try {
    const tsConfigContent = readFileSync(tsConfigPath, "utf-8");
    const cleanTsConfigContent = stripJsonComments(tsConfigContent);
    return JSON.parse(cleanTsConfigContent);
  } catch {
    return null;
  }
}
var tsConfigRule = {
  name: "ts-config",
  description: "Checks if TypeScript config is properly configured for Mastra packages",
  async run(context) {
    const tsConfig = readTsConfig(context.rootDir);
    if (!tsConfig) {
      logger.warn("No tsconfig.json found. This might cause issues with Mastra packages.");
      return true;
    }
    const { module, moduleResolution } = tsConfig.compilerOptions || {};
    const isValidConfig = moduleResolution === "bundler" || module === "CommonJS";
    if (!isValidConfig) {
      logger.error("tsconfig.json has invalid configuration");
      logger.error("Please set either:");
      logger.error('  "compilerOptions": {');
      logger.error('    "moduleResolution": "bundler"');
      logger.error("  }");
      logger.error("or");
      logger.error('  "compilerOptions": {');
      logger.error('    "module": "CommonJS"');
      logger.error("  }");
      logger.error("For the recommended TypeScript configuration, see:");
      logger.error("https://mastra.ai/en/docs/getting-started/installation#initialize-typescript");
      return false;
    }
    logger.info("TypeScript config is properly configured for Mastra packages");
    return true;
  }
};

// src/commands/lint/rules/index.ts
var rules = [nextConfigRule, tsConfigRule, mastraCoreRule];

// src/commands/lint/index.ts
function readPackageJson(dir2) {
  const packageJsonPath = join(dir2, "package.json");
  try {
    const packageJsonContent = readFileSync(packageJsonPath, "utf-8");
    return JSON.parse(packageJsonContent);
  } catch (error) {
    if (error instanceof Error) {
      logger.error("Failed to read package.json", { error: error.message });
    }
    throw error;
  }
}
function getMastraPackages2(packageJson) {
  const allDependencies = {
    ...packageJson.dependencies,
    ...packageJson.devDependencies
  };
  const mastraPackages = Object.entries(allDependencies).filter(
    ([name]) => name.startsWith("@mastra/") || name === "mastra"
  );
  return mastraPackages.map(([name, version2]) => ({
    name,
    version: version2,
    isAlpha: version2.includes("alpha")
  }));
}
async function lint({ dir: dir2, root, tools }) {
  try {
    const rootDir = root || process.cwd();
    const mastraDir = dir2 ? dir2.startsWith("/") ? dir2 : join(process.cwd(), dir2) : join(process.cwd(), "src", "mastra");
    const outputDirectory = join(rootDir, ".mastra");
    const defaultToolsPath = join(mastraDir, "tools");
    const discoveredTools = [defaultToolsPath, ...tools ?? []];
    const packageJson = readPackageJson(rootDir);
    const mastraPackages = getMastraPackages2(packageJson);
    const context = {
      rootDir,
      mastraDir,
      outputDirectory,
      discoveredTools,
      packageJson,
      mastraPackages
    };
    const results = await Promise.all(rules.map((rule) => rule.run(context)));
    const allRulesPassed = results.every((result) => result);
    if (allRulesPassed) {
      const fileService = new FileService();
      const mastraEntryFile = fileService.getFirstExistingFile([
        join(mastraDir, "index.ts"),
        join(mastraDir, "index.js")
      ]);
      const platformDeployer = await getDeployer(mastraEntryFile, outputDirectory);
      if (!platformDeployer) {
        const deployer = new BuildBundler();
        await deployer.lint(mastraEntryFile, outputDirectory, discoveredTools);
      } else {
        await platformDeployer.lint(mastraEntryFile, outputDirectory, discoveredTools);
      }
    }
    return allRulesPassed;
  } catch (error) {
    if (error instanceof Error) {
      logger.error("Lint check failed", { error: error.message });
    }
    return false;
  }
}

// src/commands/actions/lint-project.ts
var lintProject = async (args) => {
  await analytics.trackCommandExecution({
    command: "lint",
    args,
    execution: async () => {
      await lint({ dir: args.dir, root: args.root, tools: args.tools ? args.tools.split(",") : [] });
    },
    origin: origin2
  });
};
function formatCategoryName(category) {
  return category.split("-").map((word) => word.charAt(0).toUpperCase() + word.slice(1)).join(" ");
}
function formatTable(scorers) {
  if (scorers.length === 0) return "";
  const nameWidth = Math.max(4, Math.max(...scorers.map((s) => s.name.length)));
  const idWidth = Math.max(2, Math.max(...scorers.map((s) => s.id.length)));
  const typeWidth = Math.max(4, Math.max(...scorers.map((s) => s.type.length)));
  const descWidth = Math.max(11, Math.max(...scorers.map((s) => s.description.length)));
  const header = `${pc4.bold("Name".padEnd(nameWidth))} \u2502 ${pc4.bold("ID".padEnd(idWidth))} \u2502 ${pc4.bold("Type".padEnd(typeWidth))} \u2502 ${pc4.bold("Description".padEnd(descWidth))}`;
  const separator = "\u2500".repeat(nameWidth) + "\u2500\u253C\u2500" + "\u2500".repeat(idWidth) + "\u2500\u253C\u2500" + "\u2500".repeat(typeWidth) + "\u2500\u253C\u2500" + "\u2500".repeat(descWidth) + "\u2500";
  const rows = scorers.map(
    (scorer) => `${(scorer?.name ?? scorer?.id).padEnd(nameWidth)} \u2502 ${pc4.dim(scorer.id.padEnd(idWidth))} \u2502 ${scorer.type.padEnd(typeWidth)} \u2502 ${pc4.dim(scorer.description.padEnd(descWidth))}`
  );
  return [header, separator, ...rows].join("\n");
}
function listAllScorers() {
  p7.intro(pc4.inverse(" Available Scorers "));
  const groupedScorers = AVAILABLE_SCORERS.reduce(
    (acc, scorer) => {
      if (!acc[scorer.category]) {
        acc[scorer.category] = [];
      }
      acc[scorer.category].push(scorer);
      return acc;
    },
    {}
  );
  for (const [category, scorers] of Object.entries(groupedScorers)) {
    p7.log.info(`${pc4.bold(pc4.cyan(formatCategoryName(category)))} Scorers:`);
    p7.log.message(formatTable(scorers));
  }
}

// src/commands/actions/list-scorers.ts
var origin5 = process.env.MASTRA_ANALYTICS_ORIGIN;
var listScorers = async (args) => {
  await analytics.trackCommandExecution({
    command: "scorers-list",
    args,
    execution: async () => {
      return listAllScorers();
    },
    origin: origin5
  });
};
var MigrateBundler = class extends BuildBundler {
  customEnvFile;
  constructor(customEnvFile) {
    super({ studio: false });
    this.customEnvFile = customEnvFile;
  }
  getEnvFiles() {
    if (shouldSkipDotenvLoading()) {
      return Promise.resolve([]);
    }
    const possibleFiles = [".env.development", ".env.local", ".env"];
    if (this.customEnvFile) {
      possibleFiles.unshift(this.customEnvFile);
    }
    try {
      const fileService = new FileService$2();
      const envFile = fileService.getFirstExistingFile(possibleFiles);
      return Promise.resolve([envFile]);
    } catch {
    }
    return Promise.resolve([]);
  }
  async bundle(entryFile, outputDirectory, { toolsPaths, projectRoot }) {
    return this._bundle(this.getEntry(), entryFile, { outputDirectory, projectRoot }, toolsPaths);
  }
  getEntry() {
    return `
    import { mastra } from '#mastra';

    async function runMigration() {
      const storage = mastra.getStorage();

      if (!storage) {
        console.log(JSON.stringify({
          success: false,
          alreadyMigrated: false,
          duplicatesRemoved: 0,
          message: 'Storage not configured. Please configure storage in your Mastra instance.',
        }));
        process.exit(1);
      }

      // Access the observability store directly from storage.stores
      const observabilityStore = storage.stores?.observability;

      if (!observabilityStore) {
        console.log(JSON.stringify({
          success: false,
          alreadyMigrated: false,
          duplicatesRemoved: 0,
          message: 'Observability storage not configured. Migration not required.',
        }));
        process.exit(0);
      }

      // Check if the store has a migrateSpans method
      if (typeof observabilityStore.migrateSpans !== 'function') {
        console.log(JSON.stringify({
          success: false,
          alreadyMigrated: false,
          duplicatesRemoved: 0,
          message: 'Migration not supported for this storage backend.',
        }));
        process.exit(1);
      }

      try {
        // Run the migration - migrateSpans handles everything internally
        const result = await observabilityStore.migrateSpans();

        console.log(JSON.stringify({
          success: result.success,
          alreadyMigrated: result.alreadyMigrated,
          duplicatesRemoved: result.duplicatesRemoved,
          message: result.message,
        }));

        process.exit(result.success ? 0 : 1);
      } catch (error) {
        console.log(JSON.stringify({
          success: false,
          alreadyMigrated: false,
          duplicatesRemoved: 0,
          message: error instanceof Error ? error.message : 'Unknown error during migration',
        }));
        process.exit(1);
      }
    }

    runMigration();
    `;
  }
};

// src/commands/migrate/migrate.ts
async function migrate({
  dir: dir2,
  root,
  env,
  debug,
  yes
}) {
  const logger2 = createLogger(debug);
  const rootDir = root || process3.cwd();
  const mastraDir = dir2 ? dir2.startsWith("/") ? dir2 : join(process3.cwd(), dir2) : join(process3.cwd(), "src", "mastra");
  const dotMastraPath = join(rootDir, ".mastra");
  p7.intro(pc4.cyan("Mastra Storage Migration"));
  if (!yes) {
    p7.log.warn(pc4.yellow("Warning: This migration will modify your database."));
    p7.log.message("Before proceeding, please ensure you have:");
    p7.log.message("  \u2022 Created a backup of your database");
    p7.log.message("  \u2022 Tested this migration in a non-production environment");
    const confirmed = await p7.confirm({
      message: "Have you backed up your database and are ready to proceed?",
      initialValue: false
    });
    if (p7.isCancel(confirmed) || !confirmed) {
      p7.log.info("Migration cancelled. Please back up your database before running this command.");
      p7.log.message(pc4.dim("Tip: Use --yes or -y to skip this prompt in CI/automation."));
      process3.exit(0);
    }
  }
  try {
    const fileService = new FileService$1();
    const entryFile = fileService.getFirstExistingFile([join(mastraDir, "index.ts"), join(mastraDir, "index.js")]);
    const bundler = new MigrateBundler(env);
    bundler.__setLogger(logger2);
    logger2.info("Building project for migration...");
    await bundler.prepare(dotMastraPath);
    const discoveredTools = bundler.getAllToolPaths(mastraDir, []);
    await bundler.bundle(entryFile, dotMastraPath, {
      toolsPaths: discoveredTools,
      projectRoot: rootDir
    });
    logger2.info("Running migration...");
    logger2.info(pc4.dim("This may take a while for large tables."));
    logger2.info("");
    const loadedEnv = await bundler.loadEnvVars();
    const migrationProcess = execa(process3.execPath, [join(dotMastraPath, "output", "index.mjs")], {
      cwd: rootDir,
      env: {
        NODE_ENV: "production",
        MASTRA_DISABLE_STORAGE_INIT: "true",
        // Prevent MIGRATION_REQUIRED error during import
        ...Object.fromEntries(loadedEnv)
      },
      stdio: ["inherit", "pipe", "pipe"],
      reject: false
    });
    let stdoutData = "";
    let stderrData = "";
    migrationProcess.stdout?.on("data", (data) => {
      stdoutData += data.toString();
    });
    migrationProcess.stderr?.on("data", (data) => {
      stderrData += data.toString();
      if (debug) {
        process3.stderr.write(data);
      }
    });
    const processResult = await migrationProcess;
    let result;
    try {
      const jsonMatch = stdoutData.match(/\{[^{}]*"success"[^{}]*\}/g);
      if (jsonMatch && jsonMatch.length > 0) {
        result = JSON.parse(jsonMatch[jsonMatch.length - 1]);
      }
    } catch {
    }
    if (result) {
      if (result.success) {
        if (result.alreadyMigrated) {
          logger2.info(pc4.green("\u2713 Migration already complete."));
        } else {
          logger2.info(pc4.green("\u2713 Migration completed successfully!"));
          if (result.duplicatesRemoved > 0) {
            logger2.info("Removed duplicate entries", { count: result.duplicatesRemoved });
          }
        }
        logger2.info(result.message);
      } else {
        logger2.error(pc4.red("\u2717 Migration failed."));
        logger2.error(result.message);
        process3.exit(1);
      }
    } else {
      if (processResult.exitCode !== 0) {
        logger2.error(pc4.red("\u2717 Migration failed."));
        if (stderrData) {
          logger2.error(stderrData);
        }
        if (stdoutData && !stdoutData.includes('"success"')) {
          logger2.error(stdoutData);
        }
        process3.exit(1);
      } else {
        logger2.info(pc4.green("\u2713 Migration completed."));
      }
    }
  } catch (error) {
    if (error.code === "ERR_MODULE_NOT_FOUND" || error.message?.includes("Cannot find module")) {
      logger2.error(pc4.red("Error: Could not find Mastra entry file."));
      logger2.info("");
      logger2.info("Make sure you have a mastra directory with an index.ts or index.js file.");
      logger2.info("Expected location", { path: mastraDir });
      logger2.info("");
      logger2.info("You can specify a custom directory:");
      logger2.info(pc4.cyan("  npx mastra migrate --dir path/to/mastra"));
    } else {
      logger2.error(pc4.red(`Error: ${error.message}`));
      if (debug) {
        logger2.error(error);
      }
    }
    process3.exit(1);
  }
}

// src/commands/actions/migrate.ts
var migrate2 = async (args) => {
  await analytics.trackCommandExecution({
    command: "mastra migrate",
    args: { ...args },
    execution: async () => {
      await migrate({
        dir: args?.dir,
        root: args?.root,
        env: args?.env,
        debug: args?.debug ?? false,
        yes: args?.yes ?? false
      });
    },
    origin: origin2
  });
};
var DevLogger = class {
  options;
  constructor(options = {}) {
    this.options = {
      timestamp: false,
      colors: true,
      ...options
    };
  }
  formatTime() {
    if (!this.options.timestamp) return "";
    return pc4.dim((/* @__PURE__ */ new Date()).toLocaleTimeString());
  }
  formatPrefix(text3, color2) {
    const time = this.formatTime();
    const prefix = pc4.bold(color2(text3));
    return time ? `${time} ${prefix}` : prefix;
  }
  info(message) {
    const prefix = this.formatPrefix("\u25D0", pc4.cyan);
    console.info(`${prefix} ${message}`);
  }
  success(message) {
    const prefix = this.formatPrefix("\u2713", pc4.green);
    console.info(`${prefix} ${pc4.green(message)}`);
  }
  warn(message) {
    const prefix = this.formatPrefix("\u26A0", pc4.yellow);
    console.info(`${prefix} ${pc4.yellow(message)}`);
  }
  error(message) {
    const prefix = this.formatPrefix("\u2717", pc4.red);
    console.info(`${prefix} ${pc4.red(message)}`);
  }
  starting() {
    const prefix = this.formatPrefix("\u25C7", pc4.blue);
    console.info(`${prefix} ${pc4.blue("Starting Mastra dev server...")}`);
  }
  ready(host, port, studioBasePath, apiPrefix, startTime, https) {
    let protocol = "http";
    if (https && https.key && https.cert) {
      protocol = "https";
    }
    console.info("");
    const timing = startTime ? `${Date.now() - startTime} ms` : "XXX ms";
    console.info(pc4.inverse(pc4.green(" mastra ")) + ` ${pc4.green(version)} ${pc4.gray("ready in")} ${timing}`);
    console.info("");
    console.info(`${pc4.dim("\u2502")} ${pc4.bold("Studio:")} ${pc4.cyan(`${protocol}://${host}:${port}${studioBasePath}`)}`);
    console.info(`${pc4.dim("\u2502")} ${pc4.bold("API:")}    ${`${protocol}://${host}:${port}${apiPrefix}`}`);
    console.info("");
  }
  bundling() {
    const prefix = this.formatPrefix("\u25D0", pc4.magenta);
    console.info(`${prefix} ${pc4.magenta("Bundling...")}`);
  }
  bundleComplete() {
    const prefix = this.formatPrefix("\u2713", pc4.green);
    console.info(`${prefix} ${pc4.green("Bundle complete")}`);
  }
  watching() {
    const time = this.formatTime();
    const icon = pc4.dim("\u25EF");
    const message = pc4.dim("watching for file changes...");
    const fullMessage = `${icon} ${message}`;
    console.info(time ? `${time} ${fullMessage}` : fullMessage);
  }
  restarting() {
    const prefix = this.formatPrefix("\u21BB", pc4.blue);
    console.info(`${prefix} ${pc4.blue("Restarting server...")}`);
  }
  fileChange(file) {
    const prefix = this.formatPrefix("\u26A1", pc4.cyan);
    const fileName = path.basename(file);
    console.info(`${prefix} ${pc4.cyan("File changed:")} ${pc4.dim(fileName)}`);
  }
  // Enhanced error reporting
  serverError(error) {
    console.info("");
    console.info(pc4.red("  \u2717 ") + pc4.bold(pc4.red("Server Error")));
    console.info("");
    console.info(`  ${pc4.red("\u2502")} ${error}`);
    console.info("");
  }
  shutdown() {
    console.info("");
    const prefix = this.formatPrefix("\u2713", pc4.green);
    console.info(`${prefix} ${pc4.green("Dev server stopped")}`);
  }
  envInfo(info) {
    console.info("");
    console.info(`  ${pc4.dim("\u2502")} ${pc4.bold("Environment:")} ${pc4.cyan(info.env || "development")}`);
    console.info(`  ${pc4.dim("\u2502")} ${pc4.bold("Root:")} ${pc4.dim(info.root)}`);
    console.info(`  ${pc4.dim("\u2502")} ${pc4.bold("Port:")} ${pc4.cyan(info.port.toString())}`);
  }
  raw(message) {
    console.info(message);
  }
  debug(message) {
    if (process.env.DEBUG || process.env.MASTRA_DEBUG) {
      const prefix = this.formatPrefix("\u25E6", pc4.gray);
      console.info(`${prefix} ${pc4.gray(message)}`);
    }
  }
  spinnerChars = ["\u25D0", "\u25D3", "\u25D1", "\u25D2"];
  spinnerIndex = 0;
  getSpinnerChar() {
    const char = this.spinnerChars[this.spinnerIndex];
    this.spinnerIndex = (this.spinnerIndex + 1) % this.spinnerChars.length;
    return char || "\u25D0";
  }
  clearLine() {
    process.stdout.write("\r\x1B[K");
  }
  update(message) {
    this.clearLine();
    const prefix = this.formatPrefix(this.getSpinnerChar(), pc4.cyan);
    process.stdout.write(`${prefix} ${message}`);
  }
};
var devLogger = new DevLogger();
async function loadAndValidatePresets(presetsPath) {
  const absolutePath = resolve(process.cwd(), presetsPath);
  if (!existsSync(absolutePath)) {
    throw new Error(`Presets file not found: ${absolutePath}`);
  }
  const content = await readFile(absolutePath, "utf-8");
  let parsed;
  try {
    parsed = JSON.parse(content);
  } catch {
    throw new Error(`Invalid JSON in presets file: ${presetsPath}`);
  }
  if (typeof parsed !== "object" || parsed === null || Array.isArray(parsed)) {
    throw new Error(`Presets file must contain a JSON object with named presets`);
  }
  for (const [key, value] of Object.entries(parsed)) {
    if (typeof value !== "object" || value === null || Array.isArray(value)) {
      throw new Error(`Preset "${key}" must be a JSON object`);
    }
  }
  return content;
}
function escapeJsonForHtml(json) {
  return json.replace(/\\/g, "\\\\").replace(/'/g, "\\'").replace(/\n/g, "\\n").replace(/\r/g, "\\r").replace(/</g, "\\u003c").replace(/>/g, "\\u003e").replace(/\u2028/g, "\\u2028").replace(/\u2029/g, "\\u2029");
}
var DevBundler = class extends Bundler {
  customEnvFile;
  constructor(customEnvFile) {
    super("Dev");
    this.customEnvFile = customEnvFile;
    this.platform = process.versions?.bun ? "neutral" : "node";
  }
  getEnvFiles() {
    if (shouldSkipDotenvLoading()) {
      return Promise.resolve([]);
    }
    const possibleFiles = [".env.development", ".env.local", ".env"];
    if (this.customEnvFile) {
      possibleFiles.unshift(this.customEnvFile);
    }
    try {
      const fileService = new FileService$1();
      const envFile = fileService.getFirstExistingFile(possibleFiles);
      return Promise.resolve([envFile]);
    } catch {
    }
    return Promise.resolve([]);
  }
  async prepare(outputDirectory) {
    await super.prepare(outputDirectory);
    const __filename2 = fileURLToPath(import.meta.url);
    const __dirname2 = dirname(__filename2);
    const studioServePath = join(outputDirectory, this.outputDir, "studio");
    await fsExtra.copy(join(dirname(__dirname2), join("dist", "studio")), studioServePath, {
      overwrite: true
    });
  }
  async watch(entryFile, outputDirectory, toolsPaths) {
    const __filename2 = fileURLToPath(import.meta.url);
    const __dirname2 = dirname(__filename2);
    const envFiles = await this.getEnvFiles();
    const bundlerOptions = await this.getUserBundlerOptions(entryFile, outputDirectory);
    const sourcemapEnabled = !!bundlerOptions?.sourcemap;
    const inputOptions = await getWatcherInputOptions(
      entryFile,
      this.platform,
      {
        "process.env.NODE_ENV": JSON.stringify(process.env.NODE_ENV || "development")
      },
      { sourcemap: sourcemapEnabled }
    );
    const toolsInputOptions = await this.listToolsInputOptions(toolsPaths);
    const outputDir = join(outputDirectory, this.outputDir);
    await this.writePackageJson(outputDir, /* @__PURE__ */ new Map(), {});
    const watcher = await createWatcher(
      {
        ...inputOptions,
        logLevel: inputOptions.logLevel === "silent" ? "warn" : inputOptions.logLevel,
        onwarn: (warning) => {
          if (warning.code === "CIRCULAR_DEPENDENCY") {
            if (warning.ids?.[0]?.includes("node_modules")) {
              return;
            }
            this.logger.warn("Circular dependency found", {
              dependency: warning.message.replace("Circular dependency: ", "")
            });
          }
        },
        plugins: [
          ...inputOptions.plugins,
          {
            name: "env-watcher",
            buildStart() {
              for (const envFile of envFiles) {
                this.addWatchFile(envFile);
              }
            }
          },
          {
            name: "tools-watcher",
            async buildEnd() {
              const toolImports = [];
              const toolsExports = [];
              Array.from(Object.keys(toolsInputOptions || {})).filter((key) => key.startsWith("tools/")).forEach((key, index) => {
                const toolExport = `tool${index}`;
                toolImports.push(`import * as ${toolExport} from './${key}.mjs';`);
                toolsExports.push(toolExport);
              });
              await writeFile(
                join(outputDir, "tools.mjs"),
                `${toolImports.join("\n")}

                export const tools = [${toolsExports.join(", ")}]`
              );
            }
          }
        ],
        input: {
          index: join(__dirname2, "templates", "dev.entry.js"),
          ...toolsInputOptions
        }
      },
      {
        dir: outputDir,
        sourcemap: sourcemapEnabled
      }
    );
    devLogger.info("Preparing development environment...");
    return new Promise((resolve5, reject) => {
      const cb = (event) => {
        if (event.code === "BUNDLE_END") {
          devLogger.success("Initial bundle complete");
          watcher.off("event", cb);
          resolve5(watcher);
        }
        if (event.code === "ERROR") {
          console.info(event);
          devLogger.error("Bundling failed - check console for details");
          watcher.off("event", cb);
          reject(event);
        }
      };
      watcher.on("event", cb);
    });
  }
  async bundle() {
  }
};

// src/commands/dev/dev.ts
var currentServerProcess;
var isRestarting = false;
var serverStartTime;
var requestContextPresetsJson;
var ON_ERROR_MAX_RESTARTS = 3;
var restartAllActiveWorkflowRuns = async ({ host, port }) => {
  try {
    await fetch(`http://${host}:${port}/__restart-active-workflow-runs`, {
      method: "POST"
    });
  } catch (error) {
    devLogger.error(`Failed to restart all active workflow runs: ${error}`);
    await new Promise((resolve5) => setTimeout(resolve5, 1500));
    try {
      await fetch(`http://${host}:${port}/__restart-active-workflow-runs`, {
        method: "POST"
      });
    } catch {
    }
  }
};
var startServer = async (dotMastraPath, { port, host, studioBasePath, apiPrefix, publicDir }, env, startOptions = {}, errorRestartCount = 0) => {
  let serverIsReady = false;
  try {
    serverStartTime = Date.now();
    devLogger.starting();
    const commands = [];
    const inspect = startOptions.inspect === "" ? true : startOptions.inspect;
    const inspectBrk = startOptions.inspectBrk === "" ? true : startOptions.inspectBrk;
    if (inspect) {
      const inspectFlag = typeof inspect === "string" ? `--inspect=${inspect}` : "--inspect";
      commands.push(inspectFlag);
    }
    if (inspectBrk) {
      const inspectBrkFlag = typeof inspectBrk === "string" ? `--inspect-brk=${inspectBrk}` : "--inspect-brk";
      commands.push(inspectBrkFlag);
    }
    if (startOptions.customArgs) {
      commands.push(...startOptions.customArgs);
    }
    commands.push(join(dotMastraPath, "index.mjs"));
    const packagesFilePath = join(dotMastraPath, "..", "mastra-packages.json");
    await mkdir(dotMastraPath, { recursive: true });
    if (startOptions.mastraPackages) {
      await writeFile(packagesFilePath, JSON.stringify(startOptions.mastraPackages), "utf-8");
    }
    await mkdir(publicDir, { recursive: true });
    currentServerProcess = execa(process3.execPath, commands, {
      cwd: publicDir,
      env: {
        NODE_ENV: "production",
        ...Object.fromEntries(env),
        MASTRA_DEV: "true",
        PORT: port.toString(),
        MASTRA_PACKAGES_FILE: packagesFilePath,
        ...startOptions?.https ? {
          MASTRA_HTTPS_KEY: startOptions.https.key.toString("base64"),
          MASTRA_HTTPS_CERT: startOptions.https.cert.toString("base64")
        } : {}
      },
      stdio: ["inherit", "pipe", "pipe", "ipc"],
      reject: false
    });
    if (currentServerProcess?.exitCode && currentServerProcess?.exitCode !== 0) {
      if (!currentServerProcess) {
        throw new Error(`Server failed to start`);
      }
      throw new Error(
        `Server failed to start with error: ${currentServerProcess.stderr || currentServerProcess.stdout}`
      );
    }
    if (currentServerProcess.stdout) {
      currentServerProcess.stdout.on("data", (data) => {
        const output = data.toString();
        if (!output.includes("Studio available") && !output.includes("\u{1F468}\u200D\u{1F4BB}") && !output.includes("Mastra API running on ")) {
          process3.stdout.write(output);
        }
      });
    }
    if (currentServerProcess.stderr) {
      currentServerProcess.stderr.on("data", (data) => {
        const output = data.toString();
        if (!output.includes("Studio available") && !output.includes("\u{1F468}\u200D\u{1F4BB}") && !output.includes("Mastra API running on ")) {
          process3.stderr.write(output);
        }
      });
    }
    currentServerProcess.on("error", (err) => {
      if (err.code !== "EPIPE") {
        throw err;
      }
    });
    currentServerProcess.on("exit", (code) => {
      if (code !== null && code !== 0) {
        const updateCommand = getUpdateCommand(startOptions.peerDepMismatches ?? []);
        if (updateCommand) {
          console.warn();
          devLogger.warn(`This error may be caused by mismatched package versions. Try running:`);
          console.warn(`  ${pc4.cyan(updateCommand)}`);
          console.warn();
        }
      }
    });
    currentServerProcess.on("message", async (message) => {
      if (message?.type === "server-ready") {
        serverIsReady = true;
        devLogger.ready(host, port, studioBasePath, apiPrefix, serverStartTime, startOptions.https);
        devLogger.watching();
        await restartAllActiveWorkflowRuns({ host, port });
        try {
          await fetch(`http://${host}:${port}${studioBasePath}/__refresh`, {
            method: "POST",
            headers: {
              "Content-Type": "application/json"
            }
          });
        } catch {
          await new Promise((resolve5) => setTimeout(resolve5, 1500));
          try {
            await fetch(`http://${host}:${port}${studioBasePath}/__refresh`, {
              method: "POST",
              headers: {
                "Content-Type": "application/json"
              }
            });
          } catch {
          }
        }
      }
    });
  } catch (err) {
    const execaError = err;
    if (execaError.stderr) {
      devLogger.serverError(execaError.stderr);
      devLogger.debug(`Server error output: ${execaError.stderr}`);
    }
    if (execaError.stdout) devLogger.debug(`Server output: ${execaError.stdout}`);
    const updateCommand = getUpdateCommand(startOptions.peerDepMismatches ?? []);
    if (updateCommand) {
      devLogger.warn(`This error may be caused by mismatched package versions. Try running: ${updateCommand}`);
    }
    if (!serverIsReady) {
      throw err;
    }
    setTimeout(() => {
      if (!isRestarting) {
        errorRestartCount++;
        if (errorRestartCount > ON_ERROR_MAX_RESTARTS) {
          devLogger.error(`Server failed to start after ${ON_ERROR_MAX_RESTARTS} error attempts. Giving up.`);
          process3.exit(1);
        }
        devLogger.warn(
          `Attempting to restart server after error... (Attempt ${errorRestartCount}/${ON_ERROR_MAX_RESTARTS})`
        );
        startServer(
          dotMastraPath,
          {
            port,
            host,
            studioBasePath,
            apiPrefix,
            publicDir
          },
          env,
          startOptions,
          errorRestartCount
        );
      }
    }, 1e3);
  }
};
async function checkAndRestart(dotMastraPath, { port, host, studioBasePath, apiPrefix, publicDir }, bundler, startOptions = {}) {
  if (isRestarting) {
    return;
  }
  try {
    const response = await fetch(`http://${host}:${port}${studioBasePath}/__hot-reload-status`);
    if (response.ok) {
      const status = await response.json();
      if (status.disabled) {
        devLogger.info("[Mastra Dev] - \u23F8\uFE0F  Server restart skipped: agent builder action in progress");
        return;
      }
    }
  } catch (error) {
    devLogger.debug(`[Mastra Dev] - Could not check hot reload status: ${error}`);
  }
  devLogger.info("[Mastra Dev] - \u2705 Restarting server...");
  await rebundleAndRestart(dotMastraPath, { port, host, studioBasePath, apiPrefix, publicDir }, bundler, startOptions);
}
async function rebundleAndRestart(dotMastraPath, { port, host, studioBasePath, apiPrefix, publicDir }, bundler, startOptions = {}) {
  if (isRestarting) {
    return;
  }
  isRestarting = true;
  try {
    if (currentServerProcess) {
      devLogger.restarting();
      devLogger.debug("Stopping current server...");
      currentServerProcess.kill("SIGINT");
    }
    const env = await bundler.loadEnvVars();
    if (requestContextPresetsJson) {
      env.set("MASTRA_REQUEST_CONTEXT_PRESETS", requestContextPresetsJson);
    }
    for (const [key, value] of env.entries()) {
      process3.env[key] = value;
    }
    await startServer(
      join(dotMastraPath, "output"),
      {
        port,
        host,
        studioBasePath,
        apiPrefix,
        publicDir
      },
      env,
      startOptions
    );
  } finally {
    isRestarting = false;
  }
}
async function dev({
  dir: dir2,
  root,
  tools,
  env,
  inspect,
  inspectBrk,
  customArgs,
  https,
  requestContextPresets,
  debug
}) {
  const rootDir = root || process3.cwd();
  const mastraDir = dir2 ? dir2.startsWith("/") ? dir2 : join(process3.cwd(), dir2) : join(process3.cwd(), "src", "mastra");
  const dotMastraPath = join(rootDir, ".mastra");
  const fileService = new FileService$1();
  const entryFile = fileService.getFirstExistingFile([join(mastraDir, "index.ts"), join(mastraDir, "index.js")]);
  const bundler = new DevBundler(env);
  bundler.__setLogger(createLogger(debug));
  const discoveredTools = bundler.getAllToolPaths(mastraDir, tools ?? []);
  const loadedEnv = await bundler.loadEnvVars();
  requestContextPresetsJson = void 0;
  loadedEnv.delete("MASTRA_REQUEST_CONTEXT_PRESETS");
  delete process3.env.MASTRA_REQUEST_CONTEXT_PRESETS;
  for (const [key, value] of loadedEnv.entries()) {
    process3.env[key] = value;
  }
  if (requestContextPresets) {
    try {
      requestContextPresetsJson = await loadAndValidatePresets(requestContextPresets);
      loadedEnv.set("MASTRA_REQUEST_CONTEXT_PRESETS", requestContextPresetsJson);
    } catch (error) {
      devLogger.error(`Failed to load request context presets: ${error instanceof Error ? error.message : error}`);
      process3.exit(1);
    }
  }
  const serverOptions = await getServerOptions(entryFile, join(dotMastraPath, "output"));
  let portToUse = serverOptions?.port ?? process3.env.PORT;
  let hostToUse = serverOptions?.host ?? process3.env.HOST ?? "localhost";
  const studioBasePathToUse = normalizeStudioBase(serverOptions?.studioBase ?? "/");
  const apiPrefixToUse = serverOptions?.apiPrefix ?? "/api";
  if (!portToUse || isNaN(Number(portToUse))) {
    const portList = Array.from({ length: 21 }, (_, i) => 4111 + i);
    portToUse = String(
      await getPort({
        port: portList
      })
    );
  }
  let httpsOptions = void 0;
  if (https && serverOptions?.https) {
    devLogger.warn("--https flag and server.https config are both specified. Using server.https config.");
  }
  if (serverOptions?.https) {
    httpsOptions = serverOptions.https;
  } else if (https) {
    const { key, cert } = await devcert.certificateFor(serverOptions?.host ?? "localhost");
    httpsOptions = { key, cert };
  }
  const mastraPackages = await getMastraPackages(rootDir);
  const peerDepMismatches = await checkMastraPeerDeps(mastraPackages);
  logPeerDepWarnings(peerDepMismatches);
  const startOptions = {
    inspect,
    inspectBrk,
    customArgs,
    https: httpsOptions,
    mastraPackages,
    peerDepMismatches
  };
  await bundler.prepare(dotMastraPath);
  const watcher = await bundler.watch(entryFile, dotMastraPath, discoveredTools);
  await startServer(
    join(dotMastraPath, "output"),
    {
      port: Number(portToUse),
      host: hostToUse,
      studioBasePath: studioBasePathToUse,
      apiPrefix: apiPrefixToUse,
      publicDir: join(mastraDir, "public")
    },
    loadedEnv,
    startOptions
  );
  watcher.on("event", (event) => {
    if (event.code === "BUNDLE_START") {
      devLogger.bundling();
    }
    if (event.code === "BUNDLE_END") {
      devLogger.bundleComplete();
      devLogger.info("[Mastra Dev] - Bundling finished, checking if restart is allowed...");
      checkAndRestart(
        dotMastraPath,
        {
          port: Number(portToUse),
          host: hostToUse,
          studioBasePath: studioBasePathToUse,
          apiPrefix: apiPrefixToUse,
          publicDir: join(mastraDir, "public")
        },
        bundler,
        startOptions
      );
    }
  });
  const handleShutdown = async () => {
    const analytics2 = getAnalytics();
    if (analytics2 && serverStartTime) {
      const durationMs = Date.now() - serverStartTime;
      analytics2.trackEvent("cli_dev_session_end", {
        durationMs,
        durationMinutes: Math.round(durationMs / 6e4)
      });
    }
    if (analytics2) {
      await analytics2.shutdown();
    }
    devLogger.shutdown();
    if (currentServerProcess) {
      currentServerProcess.kill();
    }
    watcher.close().catch(() => {
    }).finally(() => process3.exit(0));
  };
  process3.on("SIGINT", () => {
    handleShutdown().catch(() => process3.exit(0));
  });
  process3.on("SIGTERM", () => {
    handleShutdown().catch(() => process3.exit(0));
  });
}

// src/commands/actions/start-dev-server.ts
var startDevServer = async (args) => {
  analytics.trackCommand({
    command: "dev",
    origin: origin2
  });
  dev({
    dir: args?.dir,
    root: args?.root,
    tools: args?.tools ? args.tools.split(",") : [],
    env: args?.env,
    inspect: args?.inspectBrk ? false : args?.inspect,
    inspectBrk: args?.inspectBrk,
    customArgs: args?.customArgs ? args.customArgs.split(",") : [],
    https: args?.https,
    requestContextPresets: args?.requestContextPresets,
    debug: args.debug
  }).catch((err) => {
    logger.error(err.message);
  });
};
async function start(options = {}) {
  if (!shouldSkipDotenvLoading()) {
    config({ path: [options.env || ".env.production", ".env"], quiet: true });
  }
  const outputDir = options.dir || ".mastra/output";
  try {
    const outputPath = join(process.cwd(), outputDir);
    if (!fs.existsSync(outputPath)) {
      throw new Error(`Output directory ${outputPath} does not exist`);
    }
    const commands = [];
    if (options.customArgs) {
      commands.push(...options.customArgs);
    }
    commands.push("index.mjs");
    const server = spawn(process.execPath, commands, {
      cwd: outputPath,
      stdio: ["inherit", "inherit", "pipe"],
      env: {
        ...process.env,
        NODE_ENV: "production"
      }
    });
    let stderrBuffer = "";
    server.stderr.on("data", (data) => {
      stderrBuffer += data.toString();
    });
    server.on("exit", (code) => {
      if (code !== 0 && stderrBuffer) {
        if (stderrBuffer.includes("ERR_MODULE_NOT_FOUND")) {
          const packageNameMatch = stderrBuffer.match(/Cannot find package '([^']+)'/);
          const packageName = packageNameMatch ? packageNameMatch[1] : null;
          if (!packageName) {
            logger.error(stderrBuffer.trim());
          } else {
            logger.error("Module not found while starting Mastra server", { package: packageName });
          }
        } else {
          logger.error(stderrBuffer.trim());
        }
        process.exit(code);
      }
    });
    server.on("error", (err) => {
      logger.error("Failed to start server", { error: err.message });
      process.exit(1);
    });
    process.on("SIGINT", () => {
      server.kill("SIGINT");
      process.exit(0);
    });
    process.on("SIGTERM", () => {
      server.kill("SIGTERM");
      process.exit(0);
    });
  } catch (error) {
    logger.error("Failed to start Mastra server", { error: error.message });
    process.exit(1);
  }
}

// src/commands/actions/start-project.ts
var startProject = async (args) => {
  await analytics.trackCommandExecution({
    command: "start",
    args,
    execution: async () => {
      await start({
        dir: args.dir,
        env: args.env,
        customArgs: args.customArgs ? args.customArgs.split(",") : []
      });
    },
    origin: origin2
  });
};
function normalizeBasePath(basePath) {
  const trimmed = basePath.trim();
  if (trimmed === "" || trimmed === "/") {
    return "";
  }
  let normalized = trimmed.replace(/\/+/g, "/");
  if (!normalized.startsWith("/")) {
    normalized = `/${normalized}`;
  }
  if (normalized.endsWith("/")) {
    normalized = normalized.slice(0, -1);
  }
  return normalized;
}
var __filename$1 = fileURLToPath(import.meta.url);
var __dirname$1 = dirname(__filename$1);
async function studio(options = {
  serverHost: "localhost",
  serverPort: 4111,
  serverProtocol: "http"
}) {
  config({ path: [options.env || ".env.production", ".env"], quiet: true });
  let requestContextPresetsJson2 = "";
  if (options.requestContextPresets) {
    try {
      requestContextPresetsJson2 = await loadAndValidatePresets(options.requestContextPresets);
    } catch (error) {
      logger.error("Failed to load request context presets", { error: error.message });
      process.exit(1);
    }
  }
  try {
    const distPath = join(__dirname$1, "studio");
    if (!existsSync(distPath)) {
      logger.error("Studio distribution not found", { distPath });
      process.exit(1);
    }
    const port = options.port || 3e3;
    const server = createServer(distPath, options, requestContextPresetsJson2);
    server.listen(port, () => {
      logger.info("Mastra Studio running", { url: `http://localhost:${port}` });
    });
    process.on("SIGINT", () => {
      server.close(() => {
        process.exit(0);
      });
    });
    process.on("SIGTERM", () => {
      server.close(() => {
        process.exit(0);
      });
    });
  } catch (error) {
    logger.error("Failed to start Mastra Studio", { error: error.message });
    process.exit(1);
  }
}
var createServer = (builtStudioPath, options, requestContextPresetsJson2) => {
  const indexHtmlPath = join(builtStudioPath, "index.html");
  const basePath = normalizeBasePath(process.env.MASTRA_STUDIO_BASE_PATH ?? "");
  const experimentalFeatures = process.env.EXPERIMENTAL_FEATURES === "true" ? "true" : "false";
  const experimentalUI = process.env.MASTRA_EXPERIMENTAL_UI === "true" ? "true" : "false";
  const templatesEnabled = process.env.MASTRA_TEMPLATES === "true" ? "true" : "false";
  let html = readFileSync(indexHtmlPath, "utf8").replaceAll("%%MASTRA_STUDIO_BASE_PATH%%", basePath).replaceAll("%%MASTRA_SERVER_HOST%%", options.serverHost || "localhost").replaceAll("%%MASTRA_SERVER_PORT%%", String(options.serverPort || 4111)).replaceAll("%%MASTRA_SERVER_PROTOCOL%%", options.serverProtocol || "http").replaceAll("%%MASTRA_API_PREFIX%%", options.serverApiPrefix || "/api").replaceAll("%%MASTRA_EXPERIMENTAL_FEATURES%%", experimentalFeatures).replaceAll("%%MASTRA_TEMPLATES%%", templatesEnabled).replaceAll("%%MASTRA_CLOUD_API_ENDPOINT%%", "").replaceAll("%%MASTRA_HIDE_CLOUD_CTA%%", "").replaceAll("%%MASTRA_TELEMETRY_DISABLED%%", process.env.MASTRA_TELEMETRY_DISABLED ?? "").replaceAll("%%MASTRA_REQUEST_CONTEXT_PRESETS%%", escapeJsonForHtml(requestContextPresetsJson2)).replaceAll("%%MASTRA_EXPERIMENTAL_UI%%", experimentalUI);
  const compressedHtml = gzipSync(Buffer.from(html));
  const server = http.createServer((req, res) => {
    const url = req.url || "/";
    const queryStart = url.indexOf("?");
    const rawPathname = queryStart >= 0 ? url.slice(0, queryStart) : url;
    const query = queryStart >= 0 ? url.slice(queryStart + 1) : "";
    const pathname = rawPathname || "/";
    const pathWithoutBase = basePath && pathname.startsWith(`${basePath}/`) ? pathname.slice(basePath.length) : pathname === basePath ? "/" : pathname;
    const isAssetsPath = /(^|\/)assets\//.test(pathWithoutBase);
    const isDistAssetsPath = /(^|\/)dist\/assets\//.test(pathWithoutBase);
    const isMastraSvg = pathWithoutBase === "/mastra.svg" || pathWithoutBase.endsWith("/mastra.svg");
    const isStaticAsset = isAssetsPath || isDistAssetsPath || isMastraSvg;
    const rawEncoding = req.headers["accept-encoding"] ?? "";
    const encodingValues = Array.isArray(rawEncoding) ? rawEncoding : [rawEncoding];
    const supportsGzip = encodingValues.flatMap((v) => v.split(",")).map((v) => v.trim().toLowerCase()).some((v) => {
      const [coding, ...params] = v.split(";").map((p9) => p9.trim());
      if (coding !== "gzip") return false;
      const q = params.find((p9) => p9.startsWith("q="));
      return q ? Number(q.slice(2)) > 0 : true;
    });
    if (!isStaticAsset) {
      if (supportsGzip) {
        res.writeHead(200, {
          "Content-Type": "text/html",
          "Content-Encoding": "gzip",
          "Content-Length": compressedHtml.length,
          Vary: "Accept-Encoding"
        });
        return res.end(compressedHtml);
      }
      res.writeHead(200, { "Content-Type": "text/html", Vary: "Accept-Encoding" });
      return res.end(html);
    }
    if (basePath && pathWithoutBase !== pathname) {
      req.url = query ? `${pathWithoutBase}?${query}` : pathWithoutBase;
    }
    return handler(req, res, {
      public: builtStudioPath
    });
  });
  return server;
};

// src/commands/actions/start-studio.ts
var startStudio = async (args) => {
  analytics.trackCommand({
    command: "studio",
    origin: origin2
  });
  await studio(args);
};

// src/commands/auth/login.ts
async function loginAction() {
  await login();
}
async function logoutAction() {
  await clearCredentials();
  console.info("\nLogged out. Credentials removed.\n");
  if (process.env.MASTRA_API_TOKEN) {
    console.warn("   Note: MASTRA_API_TOKEN is still set in your environment.\n   Unset it to fully log out.\n");
  }
}
async function listOrgsAction() {
  const token = await getToken();
  const currentOrgId = await getCurrentOrgId();
  const orgs = await fetchOrgs(token);
  if (orgs.length === 0) {
    console.info("\nNo organizations found.\n");
    return;
  }
  console.info("\nOrganizations:\n");
  for (const org of orgs) {
    const marker = org.id === currentOrgId ? " (current)" : "";
    const role = org.role ? ` [${org.role}]` : "";
    console.info(`  ${org.name}${role}${marker}`);
    console.info(`    ID: ${org.id}`);
  }
  console.info("");
}
async function switchOrgAction() {
  if (process.env.MASTRA_API_TOKEN) {
    console.error("\nCannot switch org when using MASTRA_API_TOKEN. Unset it and log in with: mastra auth login\n");
    process.exit(1);
  }
  if (process.env.MASTRA_ORG_ID) {
    console.error("\nCannot switch org when MASTRA_ORG_ID is set. Unset it to use persistent org selection.\n");
    process.exit(1);
  }
  const token = await getToken();
  const currentOrgId = await getCurrentOrgId();
  const orgs = await fetchOrgs(token);
  if (orgs.length === 0) {
    console.info("\nNo organizations found.\n");
    return;
  }
  if (orgs.length === 1) {
    console.info(`
You only have one organization: ${orgs[0].name}
`);
    return;
  }
  const selected = await p7.select({
    message: "Switch to organization",
    options: orgs.map((o) => ({
      value: o.id,
      label: `${o.name}${o.id === currentOrgId ? " (current)" : ""}`
    }))
  });
  if (p7.isCancel(selected)) {
    p7.cancel("Cancelled.");
    process.exit(0);
  }
  await setCurrentOrgId(selected);
  const org = orgs.find((o) => o.id === selected);
  console.info(`
Switched to ${org.name} (${org.id})
`);
}

// src/commands/auth/tokens.ts
async function resolveOrgId() {
  return process.env.MASTRA_ORG_ID ?? await getCurrentOrgId();
}
async function createToken(token, orgId, name) {
  const resp = await platformFetch(`${MASTRA_PLATFORM_API_URL}/v1/auth/tokens`, {
    method: "POST",
    headers: {
      ...authHeaders(token, orgId),
      "Content-Type": "application/json"
    },
    body: JSON.stringify({ name })
  });
  if (!resp.ok) {
    const text3 = await resp.text();
    throwApiError("Failed to create token", resp.status, text3);
  }
  const data = await resp.json();
  return { id: data.token.id, secret: data.secret };
}
async function createTokenAction(name) {
  const token = await getToken();
  const orgId = await resolveOrgId();
  if (!orgId) throw new Error("No organization selected. Run: mastra auth orgs switch");
  const result = await createToken(token, orgId, name);
  console.info("\nToken created successfully!\n");
  console.info(`  Name:    ${name}`);
  console.info(`  ID:      ${result.id}`);
  console.info("");
  console.info(`  Secret:  ${result.secret}`);
  console.info("");
  console.info("  Save this secret \u2014 it will not be shown again.");
  console.info("  Set it as MASTRA_API_TOKEN in your CI environment.\n");
}
async function listTokensAction() {
  const token = await getToken();
  const orgId = await resolveOrgId();
  if (!orgId) throw new Error("No organization selected. Run: mastra auth orgs switch");
  const resp = await platformFetch(`${MASTRA_PLATFORM_API_URL}/v1/auth/tokens`, {
    headers: authHeaders(token, orgId)
  });
  if (!resp.ok) {
    const text3 = await resp.text();
    throwApiError("Failed to list tokens", resp.status, text3);
  }
  const data = await resp.json();
  if (data.tokens.length === 0) {
    console.info("\nNo tokens found.\n");
    return;
  }
  console.info("\nTokens:\n");
  for (const t of data.tokens) {
    const lastUsed = t.lastUsedAt ? new Date(t.lastUsedAt).toLocaleDateString() : "never";
    console.info(`  ${t.name}`);
    console.info(`    ID: ${t.id}  Key: ${t.obfuscatedValue}  Last used: ${lastUsed}`);
  }
  console.info("");
}
async function revokeTokenAction(tokenId) {
  const token = await getToken();
  const orgId = await resolveOrgId();
  if (!orgId) throw new Error("No organization selected. Run: mastra auth orgs switch");
  const resp = await platformFetch(`${MASTRA_PLATFORM_API_URL}/v1/auth/tokens/${encodeURIComponent(tokenId)}`, {
    method: "DELETE",
    headers: authHeaders(token, orgId)
  });
  if (!resp.ok) {
    const text3 = await resp.text();
    throwApiError("Failed to revoke token", resp.status, text3);
  }
  console.info(`
Token ${tokenId} revoked.
`);
}

// src/commands/auth/whoami.ts
async function whoamiAction() {
  const envToken = process.env.MASTRA_API_TOKEN;
  if (envToken) {
    const orgId2 = process.env.MASTRA_ORG_ID ?? await getCurrentOrgId();
    console.info("\n  Authenticated via MASTRA_API_TOKEN");
    if (orgId2) {
      console.info(`   Org: ${orgId2}`);
    }
    console.info("");
    return;
  }
  const creds = await loadCredentials();
  if (!creds) {
    console.info("\nNot logged in. Run: mastra auth login\n");
    process.exit(1);
  }
  const orgId = await getCurrentOrgId();
  let orgName = null;
  try {
    const orgs = await fetchOrgs(creds.token);
    const match = orgs.find((o) => o.id === orgId);
    if (match) orgName = match.name;
  } catch {
  }
  console.info(`
  ${creds.user.email}`);
  console.info(`   User ID: ${creds.user.id}`);
  if (orgName) {
    console.info(`   Org: ${orgName} (${orgId})`);
  } else if (orgId) {
    console.info(`   Org: ${orgId}`);
  }
  console.info("");
}
var PROJECT_CONFIG_FILE = ".mastra-project.json";
function resolveConfigPath(dir2, configFile) {
  if (configFile) {
    return isAbsolute(configFile) ? configFile : join(dir2, configFile);
  }
  return join(dir2, PROJECT_CONFIG_FILE);
}
async function loadProjectConfig(dir2, configFile) {
  try {
    const data = await readFile(resolveConfigPath(dir2, configFile), "utf-8");
    return JSON.parse(data);
  } catch {
    return null;
  }
}
async function saveProjectConfig(dir2, config3, configFile) {
  await writeFile(resolveConfigPath(dir2, configFile), JSON.stringify(config3, null, 2) + "\n");
}

// src/commands/server/platform-api.ts
async function fetchServerProjects(token, orgId) {
  const client = createApiClient(token, orgId);
  const { data, error, response } = await client.GET("/v1/server/projects");
  if (error) {
    throwApiError("Failed to fetch server projects", response.status);
  }
  return data.projects;
}
async function createServerProject(token, orgId, name) {
  const client = createApiClient(token, orgId);
  const { data, error, response } = await client.POST("/v1/server/projects", {
    body: { name }
  });
  if (error) {
    throwApiError(`Failed to create server project \u2014 ${error.detail ?? "unknown error"}`, response.status);
  }
  return data.project;
}
async function uploadServerDeploy(token, orgId, projectId, zipBuffer, meta) {
  const client = createApiClient(token, orgId);
  const { data, error, response } = await client.POST("/v1/server/deploys", {
    body: { projectId, projectName: meta?.projectName, envVars: meta?.envVars }
  });
  if (error) {
    throwApiError("Deploy failed", response.status);
  }
  const { id, uploadUrl } = data;
  if (!uploadUrl) {
    throw new Error("No upload URL returned");
  }
  if (uploadUrl.startsWith("file://")) {
    const { writeFile: writeFile4 } = await import('fs/promises');
    const { fileURLToPath: fileURLToPath4 } = await import('url');
    await writeFile4(fileURLToPath4(uploadUrl), Buffer.from(zipBuffer));
  } else {
    const uploadResp = await fetch(uploadUrl, {
      method: "PUT",
      headers: { "Content-Type": "application/zip" },
      body: new Uint8Array(zipBuffer)
    });
    if (!uploadResp.ok) {
      throw new Error(`Artifact upload failed: ${uploadResp.status} ${uploadResp.statusText}`);
    }
  }
  const { error: completeError, response: completeResponse } = await client.POST(
    "/v1/server/deploys/{id}/upload-complete",
    { params: { path: { id } } }
  );
  if (completeError) {
    throwApiError("Upload confirmation failed", completeResponse.status);
  }
  return { id, status: "queued" };
}
async function pollServerDeploy(deployId, token, orgId, maxWaitMs = 6e5) {
  const start2 = Date.now();
  let lastStatus = "";
  let currentToken = token;
  let client = createApiClient(currentToken, orgId);
  const logAbort = new AbortController();
  pollServerLogs(deployId, currentToken, orgId, logAbort.signal).catch(() => {
  });
  try {
    while (Date.now() - start2 < maxWaitMs) {
      const { data, error, response } = await client.GET("/v1/server/deploys/{id}", {
        params: { path: { id: deployId } }
      });
      if (error) {
        if (response.status === 401) {
          currentToken = await getToken();
          client = createApiClient(currentToken, orgId);
          continue;
        }
        throwApiError("Poll failed", response.status);
      }
      if (data.status !== lastStatus) {
        lastStatus = data.status;
      }
      const terminal = ["running", "failed", "crashed", "cancelled", "stopped"];
      if (terminal.includes(data.status)) {
        return data;
      }
      await new Promise((r) => setTimeout(r, 5e3));
    }
    throw new Error("Deploy timed out");
  } finally {
    logAbort.abort();
  }
}
async function getServerProjectEnv(token, orgId, projectId) {
  const client = createApiClient(token, orgId);
  const { data, error, response } = await client.GET("/v1/server/projects/{id}/env", {
    params: { path: { id: projectId } }
  });
  if (error) {
    throwApiError("Failed to fetch environment variables", response.status);
  }
  return data.envVars;
}
async function updateServerProjectEnv(token, orgId, projectId, envVars) {
  const client = createApiClient(token, orgId);
  const { error, response } = await client.PUT("/v1/server/projects/{id}/env", {
    params: { path: { id: projectId } },
    body: { envVars }
  });
  if (error) {
    throwApiError("Failed to update environment variables", response.status);
  }
}
async function pollServerLogs(deployId, token, orgId, signal) {
  await new Promise((r) => setTimeout(r, 3e3));
  let printedBuild = 0;
  let printedDeploy = 0;
  let currentToken = token;
  let client = createApiClient(currentToken, orgId);
  while (!signal.aborted) {
    try {
      const { data, response } = await client.GET("/v1/server/deploys/{id}/logs", {
        params: { path: { id: deployId } }
      });
      if (response.status === 401) {
        currentToken = await getToken();
        client = createApiClient(currentToken, orgId);
        continue;
      }
      if (data) {
        const newBuild = data.buildLogs.slice(printedBuild);
        for (const line of newBuild) {
          process.stdout.write(`${line}
`);
        }
        printedBuild = data.buildLogs.length;
        const newDeploy = data.deployLogs.slice(printedDeploy);
        for (const line of newDeploy) {
          process.stdout.write(`${line}
`);
        }
        printedDeploy = data.deployLogs.length;
      }
    } catch {
    }
    await new Promise((r) => setTimeout(r, 5e3));
  }
}

// src/commands/server/deploy.ts
function getPackageName(projectDir) {
  try {
    const raw = execSync(`node -p "require('./package.json').name"`, {
      cwd: projectDir,
      encoding: "utf-8",
      stdio: ["pipe", "pipe", "pipe"]
    }).trim();
    return raw.startsWith("@") ? raw.split("/")[1] ?? raw : raw;
  } catch {
    return null;
  }
}
function runBuild(projectDir) {
  const localMastra = join(projectDir, "node_modules", ".bin", "mastra");
  p7.log.step("Running mastra build...");
  try {
    execSync(`"${localMastra}" build`, {
      cwd: projectDir,
      stdio: "inherit",
      env: { ...process.env, NODE_ENV: "production" }
    });
  } catch {
    throw new Error("mastra build failed");
  }
  console.info("");
}
async function zipOutput(projectDir) {
  const outputDir = join(projectDir, ".mastra", "output");
  const tmpDir = join(tmpdir(), "mastra-deploy");
  await mkdir(tmpDir, { recursive: true });
  const zipPath = join(tmpDir, `server-deploy-${Date.now()}.zip`);
  return new Promise((resolvePromise, reject) => {
    const output = createWriteStream(zipPath);
    const archive = archiver("zip", { zlib: { level: 6 } });
    output.on("close", () => resolvePromise(zipPath));
    archive.on("error", reject);
    archive.pipe(output);
    archive.glob("**", { cwd: outputDir, ignore: ["node_modules/**"] }, { prefix: ".mastra/output" });
    archive.file(join(projectDir, "package.json"), { name: "package.json" });
    void archive.finalize();
  });
}
function parseEnvFile(content) {
  const vars = {};
  for (const line of content.split("\n")) {
    const trimmed = line.trim();
    if (!trimmed || trimmed.startsWith("#")) continue;
    const eqIdx = trimmed.indexOf("=");
    if (eqIdx === -1) continue;
    let key = trimmed.slice(0, eqIdx).trim();
    if (key.startsWith("export ")) key = key.slice(7).trim();
    let value = trimmed.slice(eqIdx + 1).trim();
    if (value.startsWith('"') && value.endsWith('"') || value.startsWith("'") && value.endsWith("'")) {
      value = value.slice(1, -1);
    }
    if (key) vars[key] = value;
  }
  return vars;
}
async function readEnvVars(projectDir) {
  const vars = {};
  for (const envFile of [".env", ".env.local", ".env.production"]) {
    try {
      const content = await readFile(join(projectDir, envFile), "utf-8");
      Object.assign(vars, parseEnvFile(content));
    } catch (err) {
      if (err instanceof Error && "code" in err && err.code === "ENOENT") continue;
      throw err;
    }
  }
  return vars;
}
async function resolveOrg(token, projectConfig, flagOrg) {
  const envOrgId = process.env.MASTRA_ORG_ID;
  if (envOrgId) {
    return { orgId: envOrgId, orgName: envOrgId };
  }
  if (flagOrg) {
    const orgs2 = await fetchOrgs(token);
    const match = orgs2.find((o) => o.id === flagOrg);
    return { orgId: flagOrg, orgName: match?.name ?? flagOrg };
  }
  if (projectConfig?.organizationId) {
    const orgs2 = await fetchOrgs(token);
    const match = orgs2.find((o) => o.id === projectConfig.organizationId);
    if (match) {
      return { orgId: match.id, orgName: match.name };
    }
  }
  const currentOrgId = await getCurrentOrgId();
  const orgs = await fetchOrgs(token);
  if (currentOrgId) {
    const match = orgs.find((o) => o.id === currentOrgId);
    if (match) {
      return { orgId: match.id, orgName: match.name };
    }
  }
  if (orgs.length === 1) {
    return { orgId: orgs[0].id, orgName: orgs[0].name };
  }
  if (orgs.length === 0) {
    throw new Error(`No organizations found. Create one at ${MASTRA_STUDIO_URL}`);
  }
  const selected = await p7.select({
    message: "Select an organization",
    options: orgs.map((o) => ({ value: o.id, label: `${o.name} (${o.id})` }))
  });
  if (p7.isCancel(selected)) {
    p7.cancel("Deploy cancelled.");
    process.exit(0);
  }
  const selectedOrg = orgs.find((o) => o.id === selected);
  return { orgId: selectedOrg.id, orgName: selectedOrg.name };
}
async function resolveProject(token, orgId, projectConfig, flagProject, defaultName) {
  const envProjectId = process.env.MASTRA_PROJECT_ID;
  if (envProjectId) {
    return { projectId: envProjectId, projectName: envProjectId, projectSlug: envProjectId };
  }
  if (flagProject) {
    const projects = await fetchServerProjects(token, orgId);
    const match2 = projects.find((proj) => proj.slug === flagProject || proj.id === flagProject);
    if (match2) {
      return { projectId: match2.id, projectName: match2.name, projectSlug: match2.slug ?? match2.name };
    }
    return { projectId: flagProject, projectName: flagProject, projectSlug: flagProject };
  }
  if (projectConfig?.projectId && projectConfig.organizationId === orgId) {
    return {
      projectId: projectConfig.projectId,
      projectName: projectConfig.projectName ?? projectConfig.projectId,
      projectSlug: projectConfig.projectSlug ?? projectConfig.projectName ?? projectConfig.projectId
    };
  }
  const name = defaultName;
  if (!name) {
    throw new Error("Could not determine project name from package.json. Use --project to specify one.");
  }
  const existing = await fetchServerProjects(token, orgId);
  const match = existing.find((proj) => proj.name === name || proj.slug === name);
  if (match) {
    return { projectId: match.id, projectName: match.name, projectSlug: match.slug ?? match.name };
  }
  const project = await createServerProject(token, orgId, name);
  return { projectId: project.id, projectName: project.name, projectSlug: project.slug ?? project.name };
}
async function serverDeployAction(dir2, opts) {
  const targetDir = resolve(dir2 || process.cwd());
  const isHeadless = Boolean(process.env.MASTRA_API_TOKEN);
  const autoAccept = opts.yes ?? isHeadless;
  p7.intro("mastra server deploy");
  const packageName = getPackageName(targetDir);
  let token;
  try {
    token = await getToken();
  } catch {
    p7.log.error(`Authentication failed. Run: mastra auth login`);
    process.exit(1);
  }
  const projectConfig = await loadProjectConfig(targetDir, opts.config);
  const hasOrg = Boolean(process.env.MASTRA_ORG_ID || opts.org || projectConfig?.organizationId);
  const hasProject = Boolean(process.env.MASTRA_PROJECT_ID || opts.project || projectConfig?.projectId);
  if (isHeadless && (!hasOrg || !hasProject)) {
    throw new Error(
      "MASTRA_ORG_ID and MASTRA_PROJECT_ID (or --org/--project flags, or .mastra-project.json) are required when MASTRA_API_TOKEN is set"
    );
  }
  const { orgId, orgName } = await resolveOrg(token, projectConfig, opts.org);
  const { projectId, projectName, projectSlug } = await resolveProject(
    token,
    orgId,
    projectConfig,
    opts.project,
    packageName
  );
  const isAlreadyLinked = projectConfig?.projectId === projectId && projectConfig?.organizationId === orgId;
  if (!isAlreadyLinked) {
    p7.note(
      [`Organization:  ${orgName}`, `Project:       ${projectName}`, `Directory:     ${targetDir}`].join("\n"),
      "Deploy settings"
    );
    if (!autoAccept) {
      const confirmed = await p7.confirm({
        message: "Deploy with these settings?"
      });
      if (p7.isCancel(confirmed) || !confirmed) {
        p7.cancel("Deploy cancelled.");
        process.exit(0);
      }
    }
    await saveProjectConfig(
      targetDir,
      {
        projectId,
        projectName,
        projectSlug,
        organizationId: orgId
      },
      opts.config
    );
    p7.log.success(`Saved ${opts.config || ".mastra-project.json"}`);
  } else {
    p7.log.info(`Organization: ${orgName} (${orgId})`);
    p7.log.info(`Project: ${projectName} (${projectId})`);
  }
  const s = p7.spinner();
  runBuild(targetDir);
  const outputEntry = join(targetDir, ".mastra", "output", "index.mjs");
  try {
    await access(outputEntry);
  } catch {
    throw new Error(".mastra/output/index.mjs not found \u2014 did the build succeed?");
  }
  s.start("Zipping build artifact...");
  const zipPath = await zipOutput(targetDir);
  const zipStat = await stat(zipPath);
  const sizeKB = zipStat.size / 1024;
  const sizeLabel = sizeKB > 1024 ? `${(sizeKB / 1024).toFixed(1)}MB` : `${sizeKB.toFixed(1)}KB`;
  s.stop(`Created ${sizeLabel} archive`);
  s.start("Reading environment variables...");
  const envVars = await readEnvVars(targetDir);
  const envCount = Object.keys(envVars).length;
  if (envCount > 0) {
    s.stop(`Found ${envCount} env var(s)`);
  } else {
    s.stop("No .env file found");
  }
  s.start("Uploading...");
  const zipBuffer = await readFile(zipPath);
  const deployResult = await uploadServerDeploy(token, orgId, projectId, zipBuffer, {
    projectName,
    envVars: envCount > 0 ? envVars : void 0
  });
  s.stop(`Deploy accepted: ${deployResult.id}`);
  await rm(zipPath, { force: true });
  p7.log.step("Streaming deploy logs...");
  const finalStatus = await pollServerDeploy(deployResult.id, token, orgId);
  if (finalStatus.status === "running") {
    p7.outro(`Deploy succeeded! ${finalStatus.instanceUrl}`);
  } else if (finalStatus.status === "failed") {
    p7.log.error(`Deploy failed: ${finalStatus.error}`);
    process.exit(1);
  } else {
    p7.log.warning(`Deploy ended with status: ${finalStatus.status}`);
    process.exit(1);
  }
}
async function resolveAuth() {
  const token = await getToken();
  const orgId = process.env.MASTRA_ORG_ID ?? await getCurrentOrgId();
  if (!orgId) {
    throw new Error("No organization selected. Run: mastra auth orgs switch");
  }
  return { token, orgId };
}
async function resolveProjectId(opts) {
  const envProjectId = process.env.MASTRA_PROJECT_ID;
  if (envProjectId) return envProjectId;
  const config3 = await loadProjectConfig(process.cwd(), opts.config);
  if (!config3?.projectId) {
    throw new Error("No linked project found. Deploy first with `mastra server deploy` or set MASTRA_PROJECT_ID.");
  }
  return config3.projectId;
}
async function envListAction(opts) {
  const { token, orgId } = await resolveAuth();
  const projectId = await resolveProjectId(opts);
  const envVars = await getServerProjectEnv(token, orgId, projectId);
  const keys = Object.keys(envVars);
  if (keys.length === 0) {
    console.info("\nNo environment variables set.\n");
    return;
  }
  console.info(`
  Environment variables (${keys.length}):
`);
  for (const key of keys.sort()) {
    const val = envVars[key];
    const masked = val.length > 4 ? val.slice(0, 4) + "..." : val;
    console.info(`    ${key}=${masked}`);
  }
  console.info("");
}
async function envSetAction(key, value, opts) {
  const { token, orgId } = await resolveAuth();
  const projectId = await resolveProjectId(opts);
  const envVars = await getServerProjectEnv(token, orgId, projectId);
  envVars[key] = value;
  await updateServerProjectEnv(token, orgId, projectId, envVars);
  console.info(`
  Set ${key} successfully.
`);
}
async function envUnsetAction(key, opts) {
  const { token, orgId } = await resolveAuth();
  const projectId = await resolveProjectId(opts);
  const envVars = await getServerProjectEnv(token, orgId, projectId);
  if (!(key in envVars)) {
    console.info(`
  ${key} is not set.
`);
    return;
  }
  delete envVars[key];
  await updateServerProjectEnv(token, orgId, projectId, envVars);
  console.info(`
  Removed ${key} successfully.
`);
}
async function envImportAction(file, opts) {
  const filePath = resolve(file);
  let content;
  try {
    content = await readFile(filePath, "utf-8");
  } catch {
    throw new Error(`Could not read file: ${filePath}`);
  }
  const newVars = parseEnvFile(content);
  const newKeys = Object.keys(newVars);
  if (newKeys.length === 0) {
    console.info("\n  No variables found in file.\n");
    return;
  }
  const { token, orgId } = await resolveAuth();
  const projectId = await resolveProjectId(opts);
  const envVars = await getServerProjectEnv(token, orgId, projectId);
  Object.assign(envVars, newVars);
  await updateServerProjectEnv(token, orgId, projectId, envVars);
  console.info(`
  Imported ${newKeys.length} variable(s) from ${file}:
`);
  for (const key of newKeys.sort()) {
    console.info(`    ${key}`);
  }
  console.info("");
}

// src/commands/studio/platform-api.ts
async function fetchProjects(token, orgId) {
  const client = createApiClient(token, orgId);
  const { data, error, response } = await client.GET("/v1/studio/projects");
  if (error) {
    throwApiError("Failed to fetch projects", response.status, error.detail);
  }
  return data.projects;
}
async function createProject2(token, orgId, name) {
  const client = createApiClient(token, orgId);
  const { data, error, response } = await client.POST("/v1/studio/projects", {
    body: { name }
  });
  if (error) {
    throwApiError("Failed to create project", response.status, error.detail);
  }
  return data.project;
}
async function fetchDeployStatus(deployId, token, orgId) {
  const client = createApiClient(token, orgId);
  const { data, error, response } = await client.GET("/v1/studio/deploys/{id}", {
    params: { path: { id: deployId } }
  });
  if (error) {
    throwApiError("Failed to fetch deploy status", response.status, error.detail);
  }
  return data.deploy;
}
async function uploadDeploy(token, orgId, projectId, zipBuffer, meta) {
  const headers = {
    ...authHeaders(token, orgId),
    "Content-Type": "application/json",
    "x-project-id": projectId
  };
  if (meta?.gitBranch) headers["x-git-branch"] = meta.gitBranch;
  if (meta?.projectName) headers["x-project-name"] = meta.projectName;
  if (meta?.mastraVersion) headers["x-mastra-version"] = meta.mastraVersion;
  const createResp = await platformFetch(`${MASTRA_PLATFORM_API_URL}/v1/studio/deploys`, {
    method: "POST",
    headers,
    body: JSON.stringify({ envVars: meta?.envVars })
  });
  if (!createResp.ok) {
    const body = await createResp.json().catch(() => ({}));
    throwApiError("Deploy failed", createResp.status, body.detail);
  }
  const { deploy } = await createResp.json();
  if (deploy.uploadUrl.startsWith("file://")) {
    const { writeFile: writeFile4 } = await import('fs/promises');
    const { fileURLToPath: fileURLToPath4 } = await import('url');
    await writeFile4(fileURLToPath4(deploy.uploadUrl), Buffer.from(zipBuffer));
  } else {
    const uploadResp = await fetch(deploy.uploadUrl, {
      method: "PUT",
      headers: { "Content-Type": "application/zip" },
      body: new Uint8Array(zipBuffer)
    });
    if (!uploadResp.ok) {
      throw new Error(`Artifact upload failed: ${uploadResp.status} ${uploadResp.statusText}`);
    }
  }
  const completeResp = await platformFetch(
    `${MASTRA_PLATFORM_API_URL}/v1/studio/deploys/${deploy.id}/upload-complete`,
    {
      method: "POST",
      headers: authHeaders(token, orgId)
    }
  );
  if (!completeResp.ok) {
    const body = await completeResp.json().catch(() => ({}));
    throwApiError("Upload confirmation failed", completeResp.status, body.detail);
  }
  return deploy;
}
async function streamDeployLogs(deployId, token, orgId, signal) {
  await new Promise((r) => setTimeout(r, 2e3));
  const url = `${MASTRA_PLATFORM_API_URL}/v1/studio/deploys/${deployId}/logs/stream`;
  const resp = await platformFetch(url, {
    headers: authHeaders(token, orgId),
    signal
  });
  if (!resp.ok || !resp.body) return;
  const reader = resp.body.getReader();
  const decoder = new TextDecoder();
  let buffer = "";
  let skipNextUrlMeta = false;
  while (!signal.aborted) {
    const { done, value } = await reader.read();
    if (done) break;
    buffer += decoder.decode(value, { stream: true });
    const lines = buffer.split("\n");
    buffer = lines.pop() ?? "";
    for (const line of lines) {
      if (line.startsWith("data:")) {
        const data = line.slice(5).trim();
        if (!data) continue;
        if (data.includes("Mastra API running") || data.includes("Studio available")) {
          skipNextUrlMeta = true;
          continue;
        }
        if (skipNextUrlMeta) {
          skipNextUrlMeta = false;
          if (/^(\x1b\[\d+m)*url(\x1b\[\d+m)*:/.test(data)) continue;
        }
        process.stdout.write(`${data}
`);
      }
    }
  }
}
async function pollDeploy(deployId, token, orgId, maxWaitMs = 6e5) {
  const start2 = Date.now();
  let lastStatus = "";
  const logAbort = new AbortController();
  streamDeployLogs(deployId, token, orgId, logAbort.signal).catch(() => {
  });
  const client = createApiClient(token, orgId);
  try {
    while (Date.now() - start2 < maxWaitMs) {
      const { data, error, response } = await client.GET("/v1/studio/deploys/{id}", {
        params: { path: { id: deployId } }
      });
      if (error) {
        throwApiError("Poll failed", response.status, error.detail);
      }
      const { deploy } = data;
      if (deploy.status !== lastStatus) {
        lastStatus = deploy.status;
      }
      if (deploy.status === "running" || deploy.status === "failed" || deploy.status === "stopped") {
        return deploy;
      }
      await new Promise((r) => setTimeout(r, 2e3));
    }
    throw new Error("Deploy timed out");
  } finally {
    logAbort.abort();
  }
}

// src/commands/studio/deploy.ts
function elapsed(ms) {
  return ms < 1e3 ? `${Math.round(ms)}ms` : `${(ms / 1e3).toFixed(1)}s`;
}
function getPackageName2(projectDir) {
  try {
    const raw = execSync(`node -p "require('./package.json').name"`, {
      cwd: projectDir,
      encoding: "utf-8",
      stdio: ["pipe", "pipe", "pipe"]
    }).trim();
    return raw.startsWith("@") ? raw.split("/")[1] ?? raw : raw;
  } catch {
    return null;
  }
}
function getGitBranch(projectDir) {
  try {
    return execSync("git rev-parse --abbrev-ref HEAD", {
      cwd: projectDir,
      encoding: "utf-8",
      stdio: ["pipe", "pipe", "pipe"]
    }).trim();
  } catch {
    return null;
  }
}
function getMastraVersion(projectDir) {
  try {
    const pkgPath = join(projectDir, "package.json");
    const pkg = JSON.parse(readFileSync(pkgPath, "utf-8"));
    const deps = { ...pkg.dependencies, ...pkg.devDependencies };
    return deps["mastra"] ?? null;
  } catch {
    return null;
  }
}
function runBuild2(projectDir) {
  const localMastra = join(projectDir, "node_modules", ".bin", "mastra");
  p7.log.step("Running mastra build...");
  try {
    execSync(`"${localMastra}" build`, {
      cwd: projectDir,
      stdio: "inherit",
      env: { ...process.env, NODE_ENV: "production" }
    });
  } catch {
    throw new Error("mastra build failed");
  }
  console.info("");
}
async function zipOutput2(projectDir) {
  const outputDir = join(projectDir, ".mastra", "output");
  const tmpDir = join(tmpdir(), "mastra-deploy");
  await mkdir(tmpDir, { recursive: true });
  const zipPath = join(tmpDir, `deploy-${Date.now()}.zip`);
  return new Promise((resolvePromise, reject) => {
    const output = createWriteStream(zipPath);
    const archive = archiver("zip", { zlib: { level: 6 } });
    output.on("close", () => resolvePromise(zipPath));
    archive.on("error", reject);
    archive.pipe(output);
    archive.glob("**", { cwd: outputDir, ignore: ["node_modules/**"] }, { prefix: "output" });
    void archive.finalize();
  });
}
function parseEnvFile2(content) {
  const vars = {};
  for (const line of content.split("\n")) {
    const trimmed = line.trim();
    if (!trimmed || trimmed.startsWith("#")) continue;
    const eqIdx = trimmed.indexOf("=");
    if (eqIdx === -1) continue;
    let key = trimmed.slice(0, eqIdx).trim();
    if (key.startsWith("export ")) key = key.slice(7).trim();
    let value = trimmed.slice(eqIdx + 1).trim();
    if (value.startsWith('"') && value.endsWith('"') || value.startsWith("'") && value.endsWith("'")) {
      value = value.slice(1, -1);
    }
    if (key) vars[key] = value;
  }
  return vars;
}
async function readEnvVars2(projectDir) {
  const vars = {};
  for (const envFile of [".env.production", ".env.local", ".env"]) {
    try {
      const content = await readFile(join(projectDir, envFile), "utf-8");
      Object.assign(vars, parseEnvFile2(content));
    } catch (err) {
      if (err instanceof Error && "code" in err && err.code === "ENOENT") continue;
      throw err;
    }
  }
  return vars;
}
async function resolveOrg2(token, projectConfig, flagOrg) {
  const envOrgId = process.env.MASTRA_ORG_ID;
  if (envOrgId) {
    return { orgId: envOrgId, orgName: envOrgId };
  }
  if (flagOrg) {
    const orgs2 = await fetchOrgs(token);
    const match = orgs2.find((o) => o.id === flagOrg);
    return { orgId: flagOrg, orgName: match?.name ?? flagOrg };
  }
  if (projectConfig?.organizationId) {
    const orgs2 = await fetchOrgs(token);
    const match = orgs2.find((o) => o.id === projectConfig.organizationId);
    if (match) {
      return { orgId: match.id, orgName: match.name };
    }
  }
  const currentOrgId = await getCurrentOrgId();
  const orgs = await fetchOrgs(token);
  if (currentOrgId) {
    const match = orgs.find((o) => o.id === currentOrgId);
    if (match) {
      return { orgId: match.id, orgName: match.name };
    }
  }
  if (orgs.length === 1) {
    return { orgId: orgs[0].id, orgName: orgs[0].name };
  }
  if (orgs.length === 0) {
    throw new Error(`You have no organizations. Please create one at ${MASTRA_STUDIO_URL}`);
  }
  const selected = await p7.select({
    message: "Select an organization",
    options: orgs.map((o) => ({ value: o.id, label: `${o.name} (${o.id})` }))
  });
  if (p7.isCancel(selected)) {
    p7.cancel("Deploy cancelled.");
    process.exit(0);
  }
  const selectedOrg = orgs.find((o) => o.id === selected);
  return { orgId: selectedOrg.id, orgName: selectedOrg.name };
}
async function resolveProject2(token, orgId, projectConfig, flagProject, defaultName) {
  const envProjectId = process.env.MASTRA_PROJECT_ID;
  if (envProjectId) {
    return { existing: true, projectId: envProjectId, projectName: envProjectId, projectSlug: envProjectId };
  }
  if (flagProject) {
    const projects = await fetchProjects(token, orgId);
    const match = projects.find((proj) => proj.slug === flagProject || proj.id === flagProject);
    if (match) {
      return { existing: true, projectId: match.id, projectName: match.name, projectSlug: match.slug ?? match.name };
    }
    return { existing: false, projectName: flagProject };
  }
  if (projectConfig?.projectId && projectConfig.organizationId === orgId) {
    return {
      existing: true,
      projectId: projectConfig.projectId,
      projectName: projectConfig.projectName ?? projectConfig.projectId,
      projectSlug: projectConfig.projectSlug ?? projectConfig.projectName ?? projectConfig.projectId
    };
  }
  const name = defaultName;
  if (!name) {
    throw new Error("Could not determine project name from package.json. Use --project to specify one.");
  }
  return { existing: false, projectName: name };
}
async function deployAction(dir2, opts) {
  const targetDir = resolve(dir2 || process.cwd());
  const isHeadless = Boolean(process.env.MASTRA_API_TOKEN);
  if (isHeadless && (!process.env.MASTRA_ORG_ID || !process.env.MASTRA_PROJECT_ID)) {
    throw new Error("MASTRA_ORG_ID and MASTRA_PROJECT_ID are required when MASTRA_API_TOKEN is set");
  }
  const autoAccept = opts.yes ?? isHeadless;
  p7.intro("mastra studio deploy");
  const packageName = getPackageName2(targetDir);
  const gitBranch = getGitBranch(targetDir);
  const mastraVersion = getMastraVersion(targetDir);
  const token = await getToken();
  const projectConfig = await loadProjectConfig(targetDir, opts.config);
  const { orgId, orgName } = await resolveOrg2(token, projectConfig, opts.org);
  const resolution = await resolveProject2(token, orgId, projectConfig, opts.project, packageName);
  let projectId;
  let projectName;
  let projectSlug;
  if (resolution.existing) {
    projectId = resolution.projectId;
    projectName = resolution.projectName;
    projectSlug = resolution.projectSlug;
    const isAlreadyLinked = projectConfig?.projectId === projectId && projectConfig?.organizationId === orgId;
    p7.note(
      [
        `Organization:  ${orgName}`,
        `Project:       ${projectName}`,
        `Directory:     ${targetDir}`,
        ...gitBranch ? [`Git branch:    ${gitBranch}`] : [],
        ...mastraVersion ? [`Mastra:        ${mastraVersion}`] : []
      ].join("\n"),
      "Deploy settings"
    );
    if (!autoAccept) {
      const confirmed = await p7.confirm({
        message: "Deploy with these settings?"
      });
      if (p7.isCancel(confirmed) || !confirmed) {
        p7.cancel("Deploy cancelled.");
        process.exit(0);
      }
    }
    if (!isAlreadyLinked) {
      await saveProjectConfig(targetDir, { projectId, projectName, projectSlug, organizationId: orgId }, opts.config);
      p7.log.success(`Saved ${opts.config || ".mastra-project.json"}`);
    }
  } else {
    projectName = resolution.projectName;
    p7.note(
      [
        `Organization:  ${orgName}`,
        `Project:       ${projectName} (new)`,
        `Directory:     ${targetDir}`,
        ...gitBranch ? [`Git branch:    ${gitBranch}`] : [],
        ...mastraVersion ? [`Mastra:        ${mastraVersion}`] : []
      ].join("\n"),
      "Deploy settings"
    );
    if (!autoAccept) {
      const confirmed = await p7.confirm({
        message: "Create project and deploy?"
      });
      if (p7.isCancel(confirmed) || !confirmed) {
        p7.cancel("Deploy cancelled.");
        process.exit(0);
      }
    }
    const project = await createProject2(token, orgId, projectName);
    projectId = project.id;
    projectSlug = project.slug ?? project.name;
    p7.log.success(`Created project "${projectName}"`);
    await saveProjectConfig(targetDir, { projectId, projectName, projectSlug, organizationId: orgId }, opts.config);
    p7.log.success(`Saved ${opts.config || ".mastra-project.json"}`);
  }
  const s = p7.spinner();
  const tTotal = performance.now();
  let t;
  if (opts.skipBuild) {
    p7.log.step("Skipping build (--skip-build)");
  } else {
    t = performance.now();
    runBuild2(targetDir);
    p7.log.step(`Build completed (${elapsed(performance.now() - t)})`);
  }
  const outputEntry = join(targetDir, ".mastra", "output", "index.mjs");
  try {
    await access(outputEntry);
  } catch {
    throw new Error(".mastra/output/index.mjs not found \u2014 did the build succeed?");
  }
  t = performance.now();
  s.start("Zipping build artifact...");
  const zipPath = await zipOutput2(targetDir);
  const zipStat = await stat(zipPath);
  const sizeKB = zipStat.size / 1024;
  const sizeLabel = sizeKB > 1024 ? `${(sizeKB / 1024).toFixed(1)}MB` : `${sizeKB.toFixed(1)}KB`;
  s.stop(`Created ${sizeLabel} archive (${elapsed(performance.now() - t)})`);
  s.start("Reading environment variables...");
  const envVars = await readEnvVars2(targetDir);
  const envCount = Object.keys(envVars).length;
  if (envCount > 0) {
    s.stop(`Found ${envCount} env var(s)`);
  } else {
    s.stop("No .env file found");
  }
  t = performance.now();
  s.start("Uploading...");
  const zipBuffer = await readFile(zipPath);
  const deployResult = await uploadDeploy(token, orgId, projectId, zipBuffer, {
    gitBranch: gitBranch ?? void 0,
    projectName,
    envVars: envCount > 0 ? envVars : void 0,
    mastraVersion: mastraVersion ?? void 0
  });
  s.stop(`Uploaded (${elapsed(performance.now() - t)})`);
  await rm(zipPath, { force: true });
  p7.log.step("Streaming deploy logs...");
  const finalStatus = await pollDeploy(deployResult.id, token, orgId);
  if (finalStatus.status === "running") {
    p7.outro(`Deploy succeeded in ${elapsed(performance.now() - tTotal)}! ${finalStatus.instanceUrl}`);
  } else if (finalStatus.status === "failed") {
    p7.log.error(`Deploy failed: ${finalStatus.error}`);
    process.exit(1);
  } else {
    p7.log.warning(`Deploy ended with status: ${finalStatus.status}`);
    process.exit(1);
  }
}

// src/commands/studio/deploy-list.ts
var statusIcon = {
  starting: "\u{1F680}",
  running: "\u2705",
  stopped: "\u23F9\uFE0F",
  failed: "\u274C",
  unknown: "\u2753"
};
async function deploysAction() {
  const token = await getToken();
  const orgId = await getCurrentOrgId();
  if (!orgId) {
    console.error("No organization selected. Run: mastra auth login");
    process.exit(1);
  }
  await validateOrgAccess(token, orgId);
  const projects = await fetchProjects(token, orgId);
  if (projects.length === 0) {
    console.info("No deploys found.");
    return;
  }
  for (const project of projects) {
    const status = project.latestDeployStatus ?? "none";
    const icon = statusIcon[status] ?? "\u2753";
    const url = project.instanceUrl ?? "";
    console.info(`${icon} ${project.name} (${project.id})`);
    console.info(`   Latest:   ${project.latestDeployId ?? "none"} \u2014 ${status}`);
    if (url) {
      console.info(`   URL:      ${url}`);
    }
    console.info("");
  }
}

// src/commands/studio/deploy-logs.ts
async function getLogs(deployId, tail, token, orgId) {
  const client = createApiClient(token, orgId);
  const { data, error } = await client.GET("/v1/studio/deploys/{id}/logs", {
    params: {
      path: { id: deployId },
      query: tail ? { tail } : void 0
    }
  });
  if (error) {
    throw new Error(`Failed to fetch logs: ${error.detail}`);
  }
  if (data.logs) {
    process.stdout.write(data.logs);
  } else {
    console.info("(no logs available)");
  }
}
async function streamLogs(deployId, token, orgId) {
  const url = `${MASTRA_PLATFORM_API_URL}/v1/studio/deploys/${deployId}/logs/stream`;
  const resp = await platformFetch(url, {
    headers: {
      ...authHeaders(token, orgId),
      Accept: "text/event-stream"
    }
  });
  if (!resp.ok) {
    const text3 = await resp.text();
    throw new Error(`Failed to stream logs: ${resp.status} \u2014 ${text3}`);
  }
  if (!resp.body) {
    throw new Error("No response body for log stream");
  }
  const reader = resp.body.getReader();
  const decoder = new TextDecoder();
  let buffer = "";
  process.on("SIGINT", () => {
    void reader.cancel();
    process.exit(0);
  });
  while (true) {
    const { done, value } = await reader.read();
    if (done) break;
    buffer += decoder.decode(value, { stream: true });
    const lines = buffer.split("\n");
    buffer = lines.pop() ?? "";
    for (const line of lines) {
      if (line.startsWith("data:")) {
        const data = line.slice(5).trimStart();
        process.stdout.write(data + "\n");
      }
    }
  }
}
async function logsAction(deployId, opts) {
  const token = await getToken();
  const orgId = await getCurrentOrgId();
  if (!orgId) {
    console.error("No organization selected. Run: mastra auth login");
    process.exit(1);
  }
  await validateOrgAccess(token, orgId);
  if (opts.follow) {
    await streamLogs(deployId, token, orgId);
  } else {
    await getLogs(deployId, opts.tail, token, orgId);
  }
}

// src/commands/studio/deploy-status.ts
function printDeploy(deploy) {
  const statusIcon2 = {
    starting: "\u{1F680}",
    running: "\u2705",
    stopped: "\u23F9\uFE0F",
    failed: "\u274C",
    unknown: "\u2753"
  };
  const icon = statusIcon2[deploy.status] ?? "\u2753";
  console.info(`${icon} Deploy ${deploy.id}`);
  console.info(`   Status:   ${deploy.status}`);
  if (deploy.projectName) {
    console.info(`   Project:  ${deploy.projectName}`);
  }
  if (deploy.instanceUrl) {
    console.info(`   URL:      ${deploy.instanceUrl}`);
  }
  if (deploy.error) {
    console.info(`   Error:    ${deploy.error}`);
  }
  if (deploy.createdAt) {
    console.info(`   Created:  ${deploy.createdAt}`);
  }
}
async function statusAction(deployId, opts) {
  const token = await getToken();
  const orgId = await getCurrentOrgId();
  if (!orgId) {
    console.error("No organization selected. Run: mastra auth login");
    process.exit(1);
  }
  await validateOrgAccess(token, orgId);
  if (opts.watch) {
    let lastStatus = "";
    console.info(`Watching deploy ${deployId}...
`);
    while (true) {
      const deploy = await fetchDeployStatus(deployId, token, orgId);
      if (deploy.status !== lastStatus) {
        printDeploy(deploy);
        console.info("");
        lastStatus = deploy.status;
      }
      if (deploy.status === "running" || deploy.status === "failed" || deploy.status === "stopped") {
        break;
      }
      await new Promise((r) => setTimeout(r, 1e3));
    }
  } else {
    const deploy = await fetchDeployStatus(deployId, token, orgId);
    printDeploy(deploy);
  }
}
async function resolveCurrentOrg(token) {
  const currentOrgId = await getCurrentOrgId();
  const orgs = await fetchOrgs(token);
  if (currentOrgId) {
    const match = orgs.find((o) => o.id === currentOrgId);
    if (match) return { orgId: match.id, orgName: match.name };
  }
  if (orgs.length === 1) {
    return { orgId: orgs[0].id, orgName: orgs[0].name };
  }
  if (orgs.length === 0) {
    throw new Error("No organizations found.");
  }
  const selected = await p7.select({
    message: "Select an organization",
    options: orgs.map((o) => ({ value: o.id, label: `${o.name} (${o.id})` }))
  });
  if (p7.isCancel(selected)) {
    p7.cancel("Cancelled.");
    process.exit(0);
  }
  const org = orgs.find((o) => o.id === selected);
  return { orgId: org.id, orgName: org.name };
}
async function listProjectsAction() {
  const token = await getToken();
  const { orgId, orgName } = await resolveCurrentOrg(token);
  const projects = await fetchProjects(token, orgId);
  console.info(`
Projects in ${orgName}:
`);
  if (projects.length === 0) {
    console.info("  No projects yet. Run: mastra studio projects create\n");
    return;
  }
  for (const proj of projects) {
    const status = proj.latestDeployStatus ? ` [${proj.latestDeployStatus}]` : "";
    console.info(`  ${proj.name}${status}`);
    console.info(`    ID: ${proj.id}`);
    if (proj.instanceUrl) {
      console.info(`    URL: ${proj.instanceUrl}`);
    }
  }
  console.info("");
}
async function createProjectAction() {
  const token = await getToken();
  const { orgId, orgName } = await resolveCurrentOrg(token);
  const name = await p7.text({
    message: `Project name (in ${orgName})`,
    placeholder: "my-mastra-app",
    validate: (v) => !v || v.trim().length === 0 ? "Name is required" : void 0
  });
  if (p7.isCancel(name)) {
    p7.cancel("Cancelled.");
    process.exit(0);
  }
  const project = await createProject2(token, orgId, name);
  console.info(`
Created project: ${project.name} (${project.id})
`);
}

// src/index.ts
function wrapAction(fn) {
  return (...args) => {
    fn(...args).catch((err) => {
      console.error(`Error: ${err.message}`);
      process.exit(1);
    });
  };
}
var mastraPkg = package_default;
var version = mastraPkg.version;
var analytics = new PosthogAnalytics({
  apiKey: "phc_SBLpZVAB6jmHOct9CABq3PF0Yn5FU3G2FgT4xUr2XrT",
  host: "https://us.posthog.com",
  version
});
setAnalytics(analytics);
var program = new Command();
var origin2 = process.env.MASTRA_ANALYTICS_ORIGIN;
program.name("mastra").version(`${version}`, "-v, --version").addHelpText(
  "before",
  `
${pc4.bold(pc4.cyan("Mastra"))} is a typescript framework for building AI applications, agents, and workflows.
`
).action(() => {
  program.help();
});
program.command("create [project-name]").description("Create a new Mastra project").option("--default", "Quick start with defaults (src, OpenAI, examples)").option(
  "-c, --components <components>",
  `Comma-separated list of components (${COMPONENTS.join(", ")})`,
  parseComponents
).option("-l, --llm <model-provider>", `Default model provider (${LLMProvider.join(", ")})`, parseLlmProvider).option("-k, --llm-api-key <api-key>", "API key for the model provider").option("-e, --example", "Include example code").option("-n, --no-example", "Do not include example code").option("-t, --timeout [timeout]", "Configurable timeout for package installation, defaults to 60000 ms").option("-d, --dir <directory>", "Target directory for Mastra source code (default: src/)").option(
  "-p, --project-name <string>",
  "Project name that will be used in package.json and as the project directory name."
).option(
  "-m, --mcp <editor>",
  "MCP Server for code editor (cursor, cursor-global, windsurf, vscode, antigravity)",
  parseMcp
).option("--skills <agents>", "Install Mastra agent skills for specified agents (comma-separated)", parseSkills).option(
  "--template [template-name]",
  "Create project from a template (use template name, public GitHub URL, or leave blank to select from list)"
).action(createProject);
program.command("init").description("Initialize Mastra in your project").option("--default", "Quick start with defaults (src, OpenAI, examples)").option("-d, --dir <directory>", "Directory for Mastra files to (defaults to src/)").option(
  "-c, --components <components>",
  `Comma-separated list of components (${COMPONENTS.join(", ")})`,
  parseComponents
).option("-l, --llm <model-provider>", `Default model provider (${LLMProvider.join(", ")})`, parseLlmProvider).option("-k, --llm-api-key <api-key>", "API key for the model provider").option("-e, --example", "Include example code").option("-n, --no-example", "Do not include example code").option(
  "-m, --mcp <editor>",
  "MCP Server for code editor (cursor, cursor-global, windsurf, vscode, antigravity)",
  parseMcp
).action(initProject);
program.command("lint").description("Lint your Mastra project").option("-d, --dir <path>", "Path to your Mastra folder").option("-r, --root <path>", "Path to your root folder").option("-t, --tools <toolsDirs>", "Comma-separated list of paths to tool files to include").action(lintProject);
program.command("dev").description("Start mastra server").option("-d, --dir <dir>", "Path to your mastra folder").option("-r, --root <root>", "Path to your root folder").option("-t, --tools <toolsDirs>", "Comma-separated list of paths to tool files to include").option("-e, --env <env>", "Custom env file to include in the dev server").option(
  "-i, --inspect [host:port]",
  "Start the dev server in inspect mode (optional: [host:]port, e.g., 0.0.0.0:9229)"
).option(
  "-b, --inspect-brk [host:port]",
  "Start the dev server in inspect mode and break at the beginning of the script (optional: [host:]port)"
).option(
  "-c, --custom-args <args>",
  "Comma-separated list of custom arguments to pass to the dev server. IE: --experimental-transform-types"
).option("-s, --https", "Enable local HTTPS").option("--request-context-presets <file>", "Path to request context presets JSON file").option("--debug", "Enable debug logs", false).action(startDevServer);
program.command("build").description("Build your Mastra project").option("-d, --dir <path>", "Path to your Mastra Folder").option("-r, --root <path>", "Path to your root folder").option("-t, --tools <toolsDirs>", "Comma-separated list of paths to tool files to include").option("-s, --studio", "Bundle the studio UI with the build").option("--debug", "Enable debug logs", false).action(buildProject);
program.command("start").description("Start your built Mastra application").option("-d, --dir <path>", "Path to your built Mastra output directory (default: .mastra/output)").option("-e, --env <env>", "Custom env file to include in the start").option(
  "-c, --custom-args <args>",
  "Comma-separated list of custom arguments to pass to the Node.js process. IE: --require=newrelic"
).action(startProject);
var studioCommand = program.command("studio").description("Manage Mastra Studio").option("-p, --port <port>", "Port to run the studio on (default: 3000)").option("-e, --env <env>", "Custom env file to include in the studio").option("-h, --server-host <serverHost>", "Host of the Mastra API server (default: localhost)").option("-s, --server-port <serverPort>", "Port of the Mastra API server (default: 4111)").option("-x, --server-protocol <serverProtocol>", "Protocol of the Mastra API server (default: http)").option("--server-api-prefix <serverApiPrefix>", "API route prefix of the Mastra server (default: /api)").option("--request-context-presets <file>", "Path to request context presets JSON file").action(startStudio);
var deployCommand = studioCommand.command("deploy [dir]").description("Deploy studio").option("--org <id>", "Organization ID").option("--project <id>", "Project ID").option("-y, --yes", "Auto-accept defaults without confirmation").option("-c, --config <file>", "Project config file path (default: .mastra-project.json)").option("--skip-build", "Skip the build step and use existing .mastra/output").action(wrapAction(deployAction));
deployCommand.command("list").description("List deployed studios").action(wrapAction(deploysAction));
deployCommand.command("status <deploy-id>").description("Show deploy status").option("-w, --watch", "Watch for status changes").action(wrapAction(statusAction));
deployCommand.command("logs <deploy-id>").description("Show deploy logs").option("-f, --follow", "Stream logs in real time").option("--tail <n>", "Number of recent log lines").action(wrapAction(logsAction));
var studioProjects = studioCommand.command("projects").description("Manage studio projects").action(wrapAction(listProjectsAction));
studioProjects.command("create").description("Create a new project").action(wrapAction(createProjectAction));
program.command("migrate").description("Run database migrations to update storage schema").option("-d, --dir <path>", "Path to your Mastra folder").option("-r, --root <path>", "Path to your root folder").option("-e, --env <env>", "Custom env file to include").option("--debug", "Enable debug logs", false).option("-y, --yes", "Skip confirmation prompt (for CI/automation)").action(migrate2);
var scorersCommand = program.command("scorers").description("Manage scorers for evaluating AI outputs");
scorersCommand.command("add [scorer-name]").description("Add a new scorer to your project").option("-d, --dir <path>", "Path to your Mastra directory (default: auto-detect)").action(addScorer);
scorersCommand.command("list").description("List available scorer templates").action(listScorers);
var authCommand = program.command("auth").description("Manage authentication");
authCommand.command("login").description("Log in to Mastra").action(wrapAction(loginAction));
authCommand.command("logout").description("Log out and clear credentials").action(wrapAction(logoutAction));
authCommand.command("whoami").description("Show current user and organization").action(wrapAction(whoamiAction));
var authOrgs = authCommand.command("orgs").description("Manage organizations").action(wrapAction(listOrgsAction));
authOrgs.command("switch").description("Switch current organization").action(wrapAction(switchOrgAction));
var authTokens = authCommand.command("tokens").description("Manage API tokens").action(wrapAction(listTokensAction));
authTokens.command("create <name>").description("Create a new API token").action(wrapAction(createTokenAction));
authTokens.command("revoke <token-id>").description("Revoke an API token").action(wrapAction(revokeTokenAction));
var serverCommand = program.command("server").description("Manage Mastra Server deployments");
serverCommand.command("deploy [dir]").description("Deploy to Mastra Server").option("--org <id>", "Organization ID").option("--project <id>", "Project ID").option("-y, --yes", "Auto-accept defaults without confirmation").option("-c, --config <file>", "Project config file path (default: .mastra-project.json)").action(wrapAction(serverDeployAction));
var serverEnvCommand = serverCommand.command("env").description("Manage server environment variables");
serverEnvCommand.command("list").description("List environment variables for the linked project").option("-c, --config <file>", "Project config file path (default: .mastra-project.json)").action(wrapAction(envListAction));
serverEnvCommand.command("set <key> <value>").description("Set an environment variable").option("-c, --config <file>", "Project config file path (default: .mastra-project.json)").action(wrapAction(envSetAction));
serverEnvCommand.command("unset <key>").description("Remove an environment variable").option("-c, --config <file>", "Project config file path (default: .mastra-project.json)").action(wrapAction(envUnsetAction));
serverEnvCommand.command("import <file>").description("Import environment variables from a .env file").option("-c, --config <file>", "Project config file path (default: .mastra-project.json)").action(wrapAction(envImportAction));
program.parse(process.argv);

export { analytics, origin2 as origin, version };
