// @mastra/server/dist/chunk-37G66LYP.cjs
'use strict';

var v4 = require('zod/v4');

// src/server/schemas/tool-providers.ts
var toolProviderIdPathParams = v4.z.object({
  providerId: v4.z.string().describe("Unique identifier for the tool provider")
});
var toolSlugPathParams = toolProviderIdPathParams.extend({
  toolSlug: v4.z.string().describe("Slug identifier for the tool")
});
var listToolProviderToolsQuerySchema = v4.z.object({
  toolkit: v4.z.string().optional().describe("Filter tools by toolkit slug"),
  search: v4.z.string().optional().describe("Search tools by name or description"),
  page: v4.z.coerce.number().optional().describe("Page number for pagination"),
  perPage: v4.z.coerce.number().optional().describe("Number of items per page")
});
var paginationSchema = v4.z.object({
  total: v4.z.number().optional(),
  page: v4.z.number().optional(),
  perPage: v4.z.number().optional(),
  hasMore: v4.z.boolean()
}).optional();
var listToolProvidersResponseSchema = v4.z.object({
  providers: v4.z.array(
    v4.z.object({
      id: v4.z.string(),
      name: v4.z.string(),
      description: v4.z.string().optional()
    })
  )
});
var listToolProviderToolkitsResponseSchema = v4.z.object({
  data: v4.z.array(
    v4.z.object({
      slug: v4.z.string(),
      name: v4.z.string(),
      description: v4.z.string().optional(),
      icon: v4.z.string().optional()
    })
  ),
  pagination: paginationSchema
});
var listToolProviderToolsResponseSchema = v4.z.object({
  data: v4.z.array(
    v4.z.object({
      slug: v4.z.string(),
      name: v4.z.string(),
      description: v4.z.string().optional(),
      toolkit: v4.z.string().optional()
    })
  ),
  pagination: paginationSchema
});
var getToolProviderToolSchemaResponseSchema = v4.z.record(v4.z.string(), v4.z.unknown());

exports.getToolProviderToolSchemaResponseSchema = getToolProviderToolSchemaResponseSchema;
exports.listToolProviderToolkitsResponseSchema = listToolProviderToolkitsResponseSchema;
exports.listToolProviderToolsQuerySchema = listToolProviderToolsQuerySchema;
exports.listToolProviderToolsResponseSchema = listToolProviderToolsResponseSchema;
exports.listToolProvidersResponseSchema = listToolProvidersResponseSchema;
exports.toolProviderIdPathParams = toolProviderIdPathParams;
exports.toolSlugPathParams = toolSlugPathParams;
