// @mastra/server/dist/chunk-GIIEDO2K.js
import { conversationItemSchema } from './chunk-6NY3T42O.js';
import { getThreadByIdResponseSchema } from './chunk-6MMZVEFC.js';
import z from 'zod';

var conversationIdPathParams = z.object({
  conversationId: z.string().describe("Unique identifier for the conversation thread")
});
var createConversationBodySchema = z.object({
  agent_id: z.string().describe("Mastra agent ID used to create the conversation thread"),
  conversation_id: z.string().optional().describe("Optional conversation ID to use as the raw threadId"),
  resource_id: z.string().optional().describe("Optional resource ID to associate with the conversation"),
  title: z.string().optional(),
  metadata: z.record(z.string(), z.unknown()).optional()
});
var conversationObjectSchema = z.object({
  id: z.string(),
  object: z.literal("conversation"),
  thread: getThreadByIdResponseSchema
});
var conversationDeletedSchema = z.object({
  id: z.string(),
  object: z.literal("conversation.deleted"),
  deleted: z.literal(true)
});
var conversationItemsListSchema = z.object({
  object: z.literal("list"),
  data: z.array(conversationItemSchema),
  first_id: z.string().nullable(),
  last_id: z.string().nullable(),
  has_more: z.boolean()
});

export { conversationDeletedSchema, conversationIdPathParams, conversationItemsListSchema, conversationObjectSchema, createConversationBodySchema };
