// @mastra/server/dist/chunk-QKYHAQLX.cjs
'use strict';

var chunkHITLRKIU_cjs = require('./chunk-HITLRKIU.cjs');
var chunkVTPTMQFA_cjs = require('./chunk-VTPTMQFA.cjs');
var chunkWC4OPIB4_cjs = require('./chunk-WC4OPIB4.cjs');
var chunkZKMVCV4G_cjs = require('./chunk-ZKMVCV4G.cjs');
var chunkEES2ZZGL_cjs = require('./chunk-EES2ZZGL.cjs');
var chunkEGSCXVMR_cjs = require('./chunk-EGSCXVMR.cjs');
var chunkB34S64RC_cjs = require('./chunk-B34S64RC.cjs');
var chunk64ITUOXI_cjs = require('./chunk-64ITUOXI.cjs');
var chunkO7I5CWRX_cjs = require('./chunk-O7I5CWRX.cjs');
var web = require('stream/web');
var v4 = require('zod/v4');

// src/server/handlers/workflows.ts
var workflows_exports = {};
chunkO7I5CWRX_cjs.__export(workflows_exports, {
  CANCEL_WORKFLOW_RUN_ROUTE: () => CANCEL_WORKFLOW_RUN_ROUTE,
  CREATE_WORKFLOW_RUN_ROUTE: () => CREATE_WORKFLOW_RUN_ROUTE,
  DELETE_WORKFLOW_RUN_BY_ID_ROUTE: () => DELETE_WORKFLOW_RUN_BY_ID_ROUTE,
  GET_WORKFLOW_BY_ID_ROUTE: () => GET_WORKFLOW_BY_ID_ROUTE,
  GET_WORKFLOW_RUN_BY_ID_ROUTE: () => GET_WORKFLOW_RUN_BY_ID_ROUTE,
  LIST_WORKFLOWS_ROUTE: () => LIST_WORKFLOWS_ROUTE,
  LIST_WORKFLOW_RUNS_ROUTE: () => LIST_WORKFLOW_RUNS_ROUTE,
  OBSERVE_STREAM_LEGACY_WORKFLOW_ROUTE: () => OBSERVE_STREAM_LEGACY_WORKFLOW_ROUTE,
  OBSERVE_STREAM_WORKFLOW_ROUTE: () => OBSERVE_STREAM_WORKFLOW_ROUTE,
  RESTART_ALL_ACTIVE_WORKFLOW_RUNS_ASYNC_ROUTE: () => RESTART_ALL_ACTIVE_WORKFLOW_RUNS_ASYNC_ROUTE,
  RESTART_ALL_ACTIVE_WORKFLOW_RUNS_ROUTE: () => RESTART_ALL_ACTIVE_WORKFLOW_RUNS_ROUTE,
  RESTART_ASYNC_WORKFLOW_ROUTE: () => RESTART_ASYNC_WORKFLOW_ROUTE,
  RESTART_WORKFLOW_ROUTE: () => RESTART_WORKFLOW_ROUTE,
  RESUME_ASYNC_WORKFLOW_ROUTE: () => RESUME_ASYNC_WORKFLOW_ROUTE,
  RESUME_STREAM_WORKFLOW_ROUTE: () => RESUME_STREAM_WORKFLOW_ROUTE,
  RESUME_WORKFLOW_ROUTE: () => RESUME_WORKFLOW_ROUTE,
  START_ASYNC_WORKFLOW_ROUTE: () => START_ASYNC_WORKFLOW_ROUTE,
  START_WORKFLOW_RUN_ROUTE: () => START_WORKFLOW_RUN_ROUTE,
  STREAM_LEGACY_WORKFLOW_ROUTE: () => STREAM_LEGACY_WORKFLOW_ROUTE,
  STREAM_WORKFLOW_ROUTE: () => STREAM_WORKFLOW_ROUTE,
  TIME_TRAVEL_ASYNC_WORKFLOW_ROUTE: () => TIME_TRAVEL_ASYNC_WORKFLOW_ROUTE,
  TIME_TRAVEL_STREAM_WORKFLOW_ROUTE: () => TIME_TRAVEL_STREAM_WORKFLOW_ROUTE,
  TIME_TRAVEL_WORKFLOW_ROUTE: () => TIME_TRAVEL_WORKFLOW_ROUTE
});
async function listWorkflowsFromSystem({ mastra, workflowId }) {
  const logger = mastra.getLogger();
  if (!workflowId) {
    throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
  }
  let workflow;
  workflow = chunkWC4OPIB4_cjs.WorkflowRegistry.getWorkflow(workflowId);
  if (!workflow) {
    try {
      workflow = mastra.getWorkflowById(workflowId);
    } catch (error) {
      logger.debug("Error getting workflow, searching agents for workflow", error);
    }
  }
  if (!workflow) {
    logger.debug("Workflow not found, searching agents for workflow", { workflowId });
    const agents = mastra.listAgents();
    if (Object.keys(agents || {}).length) {
      for (const [_, agent] of Object.entries(agents)) {
        try {
          const workflows = await agent.listWorkflows();
          if (workflows[workflowId]) {
            workflow = workflows[workflowId];
            break;
          }
        } catch (error) {
          logger.debug("Error getting workflow from agent", error);
        }
      }
    }
  }
  if (!workflow) {
    throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
  }
  return { workflow };
}
var LIST_WORKFLOWS_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "GET",
  path: "/workflows",
  responseType: "json",
  queryParamSchema: v4.z.object({
    partial: v4.z.string().optional()
  }),
  responseSchema: chunkHITLRKIU_cjs.listWorkflowsResponseSchema,
  summary: "List all workflows",
  description: "Returns a list of all available workflows in the system",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: (async ({ mastra, partial }) => {
    try {
      const workflows = mastra.listWorkflows({ serialized: false });
      const isPartial = partial === "true";
      const _workflows = Object.entries(workflows).reduce((acc, [key, workflow]) => {
        acc[key] = chunkWC4OPIB4_cjs.getWorkflowInfo(workflow, isPartial);
        return acc;
      }, {});
      return _workflows;
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error getting workflows");
    }
  })
});
var GET_WORKFLOW_BY_ID_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "GET",
  path: "/workflows/:workflowId",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  responseSchema: chunkHITLRKIU_cjs.workflowInfoSchema,
  summary: "Get workflow by ID",
  description: "Returns details for a specific workflow",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: (async ({ mastra, workflowId }) => {
    try {
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      return chunkWC4OPIB4_cjs.getWorkflowInfo(workflow);
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error getting workflow");
    }
  })
});
var LIST_WORKFLOW_RUNS_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "GET",
  path: "/workflows/:workflowId/runs",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkHITLRKIU_cjs.listWorkflowRunsQuerySchema,
  responseSchema: chunkHITLRKIU_cjs.workflowRunsResponseSchema,
  summary: "List workflow runs",
  description: "Returns a paginated list of execution runs for the specified workflow",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({
    mastra,
    workflowId,
    fromDate,
    toDate,
    page,
    perPage,
    limit,
    offset,
    resourceId,
    status,
    requestContext
  }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, resourceId);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      let finalPage = page;
      let finalPerPage = perPage;
      if (finalPerPage === void 0 && limit !== void 0) {
        finalPerPage = limit;
      }
      if (finalPage === void 0 && offset !== void 0 && finalPerPage !== void 0 && finalPerPage > 0) {
        finalPage = Math.floor(offset / finalPerPage);
      }
      if (finalPerPage !== void 0 && (typeof finalPerPage !== "number" || !Number.isInteger(finalPerPage) || finalPerPage <= 0)) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "perPage must be a positive integer" });
      }
      if (finalPage !== void 0 && (!Number.isInteger(finalPage) || finalPage < 0)) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "page must be a non-negative integer" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const workflowRuns = await workflow.listWorkflowRuns({
        fromDate: fromDate ? typeof fromDate === "string" ? new Date(fromDate) : fromDate : void 0,
        toDate: toDate ? typeof toDate === "string" ? new Date(toDate) : toDate : void 0,
        perPage: finalPerPage,
        page: finalPage,
        resourceId: effectiveResourceId,
        status
      }) || {
        runs: [],
        total: 0
      };
      return workflowRuns;
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error getting workflow runs");
    }
  }
});
var GET_WORKFLOW_RUN_BY_ID_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "GET",
  path: "/workflows/:workflowId/runs/:runId",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowRunPathParams,
  queryParamSchema: chunkHITLRKIU_cjs.workflowRunResultQuerySchema,
  responseSchema: chunkHITLRKIU_cjs.workflowRunResultSchema,
  summary: "Get workflow run by ID",
  description: "Returns a workflow run with metadata and processed execution state. Use the fields query parameter to reduce payload size by requesting only specific fields (e.g., ?fields=status,result,metadata)",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, fields, withNestedWorkflows, requestContext }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Run ID is required" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const fieldList = fields ? fields.split(",").map((f) => f.trim()) : void 0;
      const run = await workflow.getWorkflowRunById(runId, {
        withNestedWorkflows: withNestedWorkflows !== "false",
        // Default to true unless explicitly 'false'
        fields: fieldList
      });
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      return run;
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error getting workflow run");
    }
  }
});
var DELETE_WORKFLOW_RUN_BY_ID_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "DELETE",
  path: "/workflows/:workflowId/runs/:runId",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowRunPathParams,
  responseSchema: chunkHITLRKIU_cjs.workflowControlResponseSchema,
  summary: "Delete workflow run by ID",
  description: "Deletes a specific workflow run by ID",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Run ID is required" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      await workflow.deleteWorkflowRunById(runId);
      return { message: "Workflow run deleted" };
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error deleting workflow run");
    }
  }
});
var CREATE_WORKFLOW_RUN_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/create-run",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.optionalRunIdSchema,
  bodySchema: chunkHITLRKIU_cjs.createWorkflowRunBodySchema,
  responseSchema: chunkHITLRKIU_cjs.createWorkflowRunResponseSchema,
  summary: "Create workflow run",
  description: "Creates a new workflow execution instance with an optional custom run ID",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, resourceId, disableScorers, requestContext }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, resourceId);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.createRun({ runId, resourceId: effectiveResourceId, disableScorers });
      return { runId: run.runId };
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error creating workflow run");
    }
  }
});
var STREAM_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/stream",
  responseType: "stream",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  bodySchema: chunkHITLRKIU_cjs.streamWorkflowBodySchema,
  summary: "Stream workflow execution",
  description: "Executes a workflow and streams the results in real-time",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, resourceId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, resourceId);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to stream workflow" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const serverCache = mastra.getServerCache();
      const run = await workflow.createRun({ runId, resourceId: effectiveResourceId });
      const result = run.stream({ ...params, requestContext });
      return result.fullStream.pipeThrough(
        new web.TransformStream({
          transform(chunk, controller) {
            if (serverCache) {
              const cacheKey = runId;
              serverCache.listPush(cacheKey, chunk).catch(() => {
              });
            }
            controller.enqueue(chunk);
          }
        })
      );
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error streaming workflow");
    }
  }
});
var RESUME_STREAM_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/resume-stream",
  responseType: "stream",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  bodySchema: chunkHITLRKIU_cjs.resumeBodySchema,
  responseSchema: chunkVTPTMQFA_cjs.streamResponseSchema,
  summary: "Resume workflow stream",
  description: "Resumes a suspended workflow execution and continues streaming results",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to resume workflow" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      const _run = await workflow.createRun({ runId, resourceId: run.resourceId });
      const serverCache = mastra.getServerCache();
      const stream = _run.resumeStream({ ...params, requestContext }).fullStream.pipeThrough(
        new web.TransformStream({
          transform(chunk, controller) {
            if (serverCache) {
              const cacheKey = runId;
              serverCache.listPush(cacheKey, chunk).catch(() => {
              });
            }
            controller.enqueue(chunk);
          }
        })
      );
      return stream;
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error resuming workflow");
    }
  }
});
var START_ASYNC_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/start-async",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.optionalRunIdSchema,
  bodySchema: chunkHITLRKIU_cjs.startAsyncWorkflowBodySchema,
  responseSchema: chunkHITLRKIU_cjs.workflowExecutionResultSchema,
  summary: "Start workflow asynchronously",
  description: "Starts a workflow execution asynchronously without streaming results",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, resourceId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, resourceId);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const _run = await workflow.createRun({ runId, resourceId: effectiveResourceId });
      const result = await _run.start({ ...params, requestContext });
      return result;
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error starting async workflow");
    }
  }
});
var START_WORKFLOW_RUN_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/start",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  bodySchema: chunkHITLRKIU_cjs.startAsyncWorkflowBodySchema,
  responseSchema: chunkHITLRKIU_cjs.workflowControlResponseSchema,
  summary: "Start specific workflow run",
  description: "Starts execution of a specific workflow run by ID",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to start run" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      const _run = await workflow.createRun({ runId, resourceId: run.resourceId });
      void _run.start({
        ...params,
        requestContext
      });
      return { message: "Workflow run started" };
    } catch (e) {
      return chunkB34S64RC_cjs.handleError(e, "Error starting workflow run");
    }
  }
});
var OBSERVE_STREAM_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/observe",
  responseType: "stream",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  responseSchema: chunkVTPTMQFA_cjs.streamResponseSchema,
  summary: "Observe workflow stream",
  description: "Observes and streams updates from an already running workflow execution",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to observe workflow stream" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      const _run = await workflow.createRun({ runId, resourceId: run.resourceId });
      const serverCache = mastra.getServerCache();
      if (!serverCache) {
        throw new chunk64ITUOXI_cjs.HTTPException(500, { message: "Server cache not found" });
      }
      const cachedRunChunks = await serverCache.listFromTo(runId, 0);
      const combinedStream = new web.ReadableStream({
        start(controller) {
          const emitCachedChunks = async () => {
            for (const chunk of cachedRunChunks) {
              controller.enqueue(chunk);
            }
          };
          const liveStream = _run.observeStream();
          const reader = liveStream.getReader();
          const pump = async () => {
            try {
              while (true) {
                const { done, value } = await reader.read();
                if (done) {
                  controller.close();
                  break;
                }
                controller.enqueue(value);
              }
            } catch (error) {
              controller.error(error);
            } finally {
              reader.releaseLock();
            }
          };
          void emitCachedChunks().then(() => {
            void pump();
          }).catch((error) => {
            controller.error(error);
          });
        }
      });
      return combinedStream;
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error observing workflow stream");
    }
  }
});
var RESUME_ASYNC_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/resume-async",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  bodySchema: chunkHITLRKIU_cjs.resumeBodySchema,
  responseSchema: chunkHITLRKIU_cjs.workflowExecutionResultSchema,
  summary: "Resume workflow asynchronously",
  description: "Resumes a suspended workflow execution asynchronously without streaming",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to resume workflow" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      const _run = await workflow.createRun({ runId, resourceId: run.resourceId });
      const result = await _run.resume({ ...params, requestContext });
      return result;
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error resuming workflow step");
    }
  }
});
var RESUME_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/resume",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  bodySchema: chunkHITLRKIU_cjs.resumeBodySchema,
  responseSchema: chunkHITLRKIU_cjs.workflowControlResponseSchema,
  summary: "Resume workflow",
  description: "Resumes a suspended workflow execution from a specific step",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to resume workflow" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      const _run = await workflow.createRun({ runId, resourceId: run.resourceId });
      void _run.resume({ ...params, requestContext });
      return { message: "Workflow run resumed" };
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error resuming workflow");
    }
  }
});
var RESTART_ASYNC_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/restart-async",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  bodySchema: chunkHITLRKIU_cjs.restartBodySchema,
  responseSchema: chunkHITLRKIU_cjs.workflowExecutionResultSchema,
  summary: "Restart workflow asynchronously",
  description: "Restarts an active workflow execution asynchronously",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to restart workflow" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      const _run = await workflow.createRun({ runId, resourceId: run.resourceId });
      const result = await _run.restart({ ...params, requestContext });
      return result;
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error restarting workflow");
    }
  }
});
var RESTART_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/restart",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  bodySchema: chunkHITLRKIU_cjs.restartBodySchema,
  responseSchema: chunkHITLRKIU_cjs.workflowControlResponseSchema,
  summary: "Restart workflow",
  description: "Restarts an active workflow execution",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to restart workflow" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      const _run = await workflow.createRun({ runId, resourceId: run.resourceId });
      void _run.restart({ ...params, requestContext });
      return { message: "Workflow run restarted" };
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error restarting workflow");
    }
  }
});
var RESTART_ALL_ACTIVE_WORKFLOW_RUNS_ASYNC_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/restart-all-active-workflow-runs-async",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  responseSchema: chunkHITLRKIU_cjs.workflowControlResponseSchema,
  summary: "Restart all active workflow runs asynchronously",
  description: "Restarts all active workflow runs asynchronously",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId }) => {
    try {
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      await workflow.restartAllActiveWorkflowRuns();
      return { message: "All active workflow runs restarted" };
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error restarting workflow");
    }
  }
});
var RESTART_ALL_ACTIVE_WORKFLOW_RUNS_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/restart-all-active-workflow-runs",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  responseSchema: chunkHITLRKIU_cjs.workflowControlResponseSchema,
  summary: "Restart all active workflow runs",
  description: "Restarts all active workflow runs",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId }) => {
    try {
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      void workflow.restartAllActiveWorkflowRuns();
      return { message: "All active workflow runs restarted" };
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error restarting workflow");
    }
  }
});
var TIME_TRAVEL_ASYNC_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/time-travel-async",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  bodySchema: chunkHITLRKIU_cjs.timeTravelBodySchema,
  responseSchema: chunkHITLRKIU_cjs.workflowExecutionResultSchema,
  summary: "Time travel workflow asynchronously",
  description: "Time travels a workflow run asynchronously without streaming",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to time travel workflow" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      const _run = await workflow.createRun({ runId, resourceId: run.resourceId });
      const result = await _run.timeTravel({ ...params, requestContext });
      return result;
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error time traveling workflow");
    }
  }
});
var TIME_TRAVEL_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/time-travel",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  bodySchema: chunkHITLRKIU_cjs.timeTravelBodySchema,
  responseSchema: chunkHITLRKIU_cjs.workflowControlResponseSchema,
  summary: "Time travel workflow",
  description: "Time travels a workflow run, starting from a specific step",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to time travel workflow" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      const _run = await workflow.createRun({ runId, resourceId: run.resourceId });
      void _run.timeTravel({ ...params, requestContext });
      return { message: "Workflow run time travel started" };
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error time traveling workflow");
    }
  }
});
var TIME_TRAVEL_STREAM_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/time-travel-stream",
  responseType: "stream",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  bodySchema: chunkHITLRKIU_cjs.timeTravelBodySchema,
  summary: "Time travel workflow stream",
  description: "Time travels a workflow run, starting from a specific step, and streams the results in real-time",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to time travel workflow stream" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const existingRun = await workflow.getWorkflowRunById(runId);
      if (!existingRun) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(existingRun, effectiveResourceId);
      const serverCache = mastra.getServerCache();
      const run = await workflow.createRun({ runId, resourceId: existingRun.resourceId });
      const result = run.timeTravelStream({ ...params, requestContext });
      return result.fullStream.pipeThrough(
        new web.TransformStream({
          transform(chunk, controller) {
            if (serverCache) {
              const cacheKey = runId;
              serverCache.listPush(cacheKey, chunk).catch(() => {
              });
            }
            controller.enqueue(chunk);
          }
        })
      );
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error time traveling workflow stream");
    }
  }
});
var CANCEL_WORKFLOW_RUN_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/runs/:runId/cancel",
  responseType: "json",
  pathParamSchema: chunkHITLRKIU_cjs.workflowRunPathParams,
  responseSchema: chunkHITLRKIU_cjs.workflowControlResponseSchema,
  summary: "Cancel workflow run",
  description: "Cancels an in-progress workflow execution",
  tags: ["Workflows"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to cancel workflow run" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      const _run = await workflow.createRun({ runId, resourceId: run.resourceId });
      await _run.cancel();
      return { message: "Workflow run cancelled" };
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error canceling workflow run");
    }
  }
});
var STREAM_LEGACY_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/stream-legacy",
  responseType: "stream",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  bodySchema: chunkHITLRKIU_cjs.streamWorkflowBodySchema,
  responseSchema: chunkVTPTMQFA_cjs.streamResponseSchema,
  summary: "[DEPRECATED] Stream workflow with legacy format",
  description: "Legacy endpoint for streaming workflow execution. Use /workflows/:workflowId/stream instead.",
  tags: ["Workflows", "Legacy"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, resourceId, requestContext, ...params }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, resourceId);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to resume workflow" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const serverCache = mastra.getServerCache();
      const run = await workflow.createRun({ runId, resourceId: effectiveResourceId });
      const result = run.streamLegacy({
        ...params,
        requestContext,
        onChunk: async (chunk) => {
          if (serverCache) {
            const cacheKey = runId;
            await serverCache.listPush(cacheKey, chunk);
          }
        }
      });
      return result.stream;
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error executing workflow");
    }
  }
});
var OBSERVE_STREAM_LEGACY_WORKFLOW_ROUTE = chunkEGSCXVMR_cjs.createRoute({
  method: "POST",
  path: "/workflows/:workflowId/observe-stream-legacy",
  responseType: "stream",
  pathParamSchema: chunkHITLRKIU_cjs.workflowIdPathParams,
  queryParamSchema: chunkEES2ZZGL_cjs.runIdSchema,
  responseSchema: chunkVTPTMQFA_cjs.streamResponseSchema,
  summary: "[DEPRECATED] Observe workflow stream with legacy format",
  description: "Legacy endpoint for observing workflow stream. Use /workflows/:workflowId/observe instead.",
  tags: ["Workflows", "Legacy"],
  requiresAuth: true,
  handler: async ({ mastra, workflowId, runId, requestContext }) => {
    try {
      const effectiveResourceId = chunkZKMVCV4G_cjs.getEffectiveResourceId(requestContext, void 0);
      if (!workflowId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "Workflow ID is required" });
      }
      if (!runId) {
        throw new chunk64ITUOXI_cjs.HTTPException(400, { message: "runId required to observe workflow stream" });
      }
      const { workflow } = await listWorkflowsFromSystem({ mastra, workflowId });
      if (!workflow) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow not found" });
      }
      const run = await workflow.getWorkflowRunById(runId);
      if (!run) {
        throw new chunk64ITUOXI_cjs.HTTPException(404, { message: "Workflow run not found" });
      }
      await chunkZKMVCV4G_cjs.validateRunOwnership(run, effectiveResourceId);
      const _run = await workflow.createRun({ runId, resourceId: run.resourceId });
      const serverCache = mastra.getServerCache();
      if (!serverCache) {
        throw new chunk64ITUOXI_cjs.HTTPException(500, { message: "Server cache not found" });
      }
      const transformStream = new web.TransformStream();
      const writer = transformStream.writable.getWriter();
      const cachedRunChunks = await serverCache.listFromTo(runId, 0);
      for (const chunk of cachedRunChunks) {
        await writer.write(chunk);
      }
      writer.releaseLock();
      const result = _run.observeStreamLegacy();
      return result.stream?.pipeThrough(transformStream);
    } catch (error) {
      return chunkB34S64RC_cjs.handleError(error, "Error observing workflow stream");
    }
  }
});

exports.CANCEL_WORKFLOW_RUN_ROUTE = CANCEL_WORKFLOW_RUN_ROUTE;
exports.CREATE_WORKFLOW_RUN_ROUTE = CREATE_WORKFLOW_RUN_ROUTE;
exports.DELETE_WORKFLOW_RUN_BY_ID_ROUTE = DELETE_WORKFLOW_RUN_BY_ID_ROUTE;
exports.GET_WORKFLOW_BY_ID_ROUTE = GET_WORKFLOW_BY_ID_ROUTE;
exports.GET_WORKFLOW_RUN_BY_ID_ROUTE = GET_WORKFLOW_RUN_BY_ID_ROUTE;
exports.LIST_WORKFLOWS_ROUTE = LIST_WORKFLOWS_ROUTE;
exports.LIST_WORKFLOW_RUNS_ROUTE = LIST_WORKFLOW_RUNS_ROUTE;
exports.OBSERVE_STREAM_LEGACY_WORKFLOW_ROUTE = OBSERVE_STREAM_LEGACY_WORKFLOW_ROUTE;
exports.OBSERVE_STREAM_WORKFLOW_ROUTE = OBSERVE_STREAM_WORKFLOW_ROUTE;
exports.RESTART_ALL_ACTIVE_WORKFLOW_RUNS_ASYNC_ROUTE = RESTART_ALL_ACTIVE_WORKFLOW_RUNS_ASYNC_ROUTE;
exports.RESTART_ALL_ACTIVE_WORKFLOW_RUNS_ROUTE = RESTART_ALL_ACTIVE_WORKFLOW_RUNS_ROUTE;
exports.RESTART_ASYNC_WORKFLOW_ROUTE = RESTART_ASYNC_WORKFLOW_ROUTE;
exports.RESTART_WORKFLOW_ROUTE = RESTART_WORKFLOW_ROUTE;
exports.RESUME_ASYNC_WORKFLOW_ROUTE = RESUME_ASYNC_WORKFLOW_ROUTE;
exports.RESUME_STREAM_WORKFLOW_ROUTE = RESUME_STREAM_WORKFLOW_ROUTE;
exports.RESUME_WORKFLOW_ROUTE = RESUME_WORKFLOW_ROUTE;
exports.START_ASYNC_WORKFLOW_ROUTE = START_ASYNC_WORKFLOW_ROUTE;
exports.START_WORKFLOW_RUN_ROUTE = START_WORKFLOW_RUN_ROUTE;
exports.STREAM_LEGACY_WORKFLOW_ROUTE = STREAM_LEGACY_WORKFLOW_ROUTE;
exports.STREAM_WORKFLOW_ROUTE = STREAM_WORKFLOW_ROUTE;
exports.TIME_TRAVEL_ASYNC_WORKFLOW_ROUTE = TIME_TRAVEL_ASYNC_WORKFLOW_ROUTE;
exports.TIME_TRAVEL_STREAM_WORKFLOW_ROUTE = TIME_TRAVEL_STREAM_WORKFLOW_ROUTE;
exports.TIME_TRAVEL_WORKFLOW_ROUTE = TIME_TRAVEL_WORKFLOW_ROUTE;
exports.workflows_exports = workflows_exports;
