// @mastra/server/dist/chunk-P72YX3EY.js
import { NEW_ROUTES, LIST_SCORES, LIST_LOGS, LIST_FEEDBACK, GET_TAGS, GET_SERVICE_NAMES, GET_SCORE_TIME_SERIES, GET_SCORE_PERCENTILES, GET_SCORE_BREAKDOWN, GET_SCORE_AGGREGATE, GET_METRIC_TIME_SERIES, GET_METRIC_PERCENTILES, GET_METRIC_NAMES, GET_METRIC_LABEL_VALUES, GET_METRIC_LABEL_KEYS, GET_METRIC_BREAKDOWN, GET_METRIC_AGGREGATE, GET_FEEDBACK_TIME_SERIES, GET_FEEDBACK_PERCENTILES, GET_FEEDBACK_BREAKDOWN, GET_FEEDBACK_AGGREGATE, GET_ENVIRONMENTS, GET_ENTITY_TYPES, GET_ENTITY_NAMES, CREATE_SCORE, CREATE_FEEDBACK } from './chunk-OOZYQF4U.js';
import { getObservabilityStore, getStorage } from './chunk-53YFG5AY.js';
import { createRoute, pickParams, wrapSchemaForQueryParams } from './chunk-NMS2SC2B.js';
import { handleError } from './chunk-P23KBWKB.js';
import { HTTPException } from './chunk-6QWQZI4Q.js';
import { __export } from './chunk-PR4QN5HX.js';
import { extractTrajectoryFromTrace, listScoresResponseSchema } from '@mastra/core/evals';
import { scoreTraces } from '@mastra/core/evals/scoreTraces';
import { dateRangeSchema, listTracesResponseSchema, tracesFilterSchema, paginationArgsSchema, tracesOrderBySchema, getTraceResponseSchema, getTraceArgsSchema, scoreTracesResponseSchema, scoreTracesRequestSchema, spanIdsSchema } from '@mastra/core/storage';
import { z } from 'zod/v4';

// src/server/handlers/observability.ts
var observability_exports = {};
__export(observability_exports, {
  CREATE_FEEDBACK: () => CREATE_FEEDBACK,
  CREATE_SCORE: () => CREATE_SCORE,
  GET_ENTITY_NAMES: () => GET_ENTITY_NAMES,
  GET_ENTITY_TYPES: () => GET_ENTITY_TYPES,
  GET_ENVIRONMENTS: () => GET_ENVIRONMENTS,
  GET_FEEDBACK_AGGREGATE: () => GET_FEEDBACK_AGGREGATE,
  GET_FEEDBACK_BREAKDOWN: () => GET_FEEDBACK_BREAKDOWN,
  GET_FEEDBACK_PERCENTILES: () => GET_FEEDBACK_PERCENTILES,
  GET_FEEDBACK_TIME_SERIES: () => GET_FEEDBACK_TIME_SERIES,
  GET_METRIC_AGGREGATE: () => GET_METRIC_AGGREGATE,
  GET_METRIC_BREAKDOWN: () => GET_METRIC_BREAKDOWN,
  GET_METRIC_LABEL_KEYS: () => GET_METRIC_LABEL_KEYS,
  GET_METRIC_LABEL_VALUES: () => GET_METRIC_LABEL_VALUES,
  GET_METRIC_NAMES: () => GET_METRIC_NAMES,
  GET_METRIC_PERCENTILES: () => GET_METRIC_PERCENTILES,
  GET_METRIC_TIME_SERIES: () => GET_METRIC_TIME_SERIES,
  GET_SCORE_AGGREGATE: () => GET_SCORE_AGGREGATE,
  GET_SCORE_BREAKDOWN: () => GET_SCORE_BREAKDOWN,
  GET_SCORE_PERCENTILES: () => GET_SCORE_PERCENTILES,
  GET_SCORE_TIME_SERIES: () => GET_SCORE_TIME_SERIES,
  GET_SERVICE_NAMES: () => GET_SERVICE_NAMES,
  GET_TAGS: () => GET_TAGS,
  GET_TRACE_ROUTE: () => GET_TRACE_ROUTE,
  GET_TRACE_TRAJECTORY_ROUTE: () => GET_TRACE_TRAJECTORY_ROUTE,
  LIST_FEEDBACK: () => LIST_FEEDBACK,
  LIST_LOGS: () => LIST_LOGS,
  LIST_SCORES: () => LIST_SCORES,
  LIST_SCORES_BY_SPAN_ROUTE: () => LIST_SCORES_BY_SPAN_ROUTE,
  LIST_TRACES_ROUTE: () => LIST_TRACES_ROUTE,
  NEW_ROUTES: () => NEW_ROUTES,
  SCORE_TRACES_ROUTE: () => SCORE_TRACES_ROUTE
});
var legacyQueryParamsSchema = z.object({
  // Old: dateRange was in pagination, now it's startedAt in filters
  dateRange: dateRangeSchema.optional(),
  // Old: name matched span names like "agent run: 'myAgent'"
  name: z.string().optional(),
  // entityType needs preprocessing to handle legacy 'workflow' value
  entityType: z.preprocess((val) => val === "workflow" ? "workflow_run" : val, z.string().optional())
});
function transformLegacyParams(params) {
  const result = { ...params };
  if (result.entityType === "workflow") {
    result.entityType = "workflow_run";
  }
  if (params.dateRange && !params.startedAt) {
    result.startedAt = params.dateRange;
    delete result.dateRange;
  }
  if (typeof params.name === "string" && !params.entityId) {
    const agentMatch = params.name.match(/^agent run: '([^']+)'$/);
    const workflowMatch = params.name.match(/^workflow run: '([^']+)'$/);
    if (agentMatch) {
      result.entityId = agentMatch[1];
      result.entityType = "agent";
    } else if (workflowMatch) {
      result.entityId = workflowMatch[1];
      result.entityType = "workflow_run";
    }
    delete result.name;
  }
  return result;
}
async function getScoresStore(mastra) {
  const storage = getStorage(mastra);
  const scores = await storage.getStore("scores");
  if (!scores) {
    throw new HTTPException(500, { message: "Scores storage domain is not available" });
  }
  return scores;
}
var LIST_TRACES_ROUTE = createRoute({
  method: "GET",
  path: "/observability/traces",
  responseType: "json",
  queryParamSchema: wrapSchemaForQueryParams(
    tracesFilterSchema.extend(paginationArgsSchema.shape).extend(tracesOrderBySchema.shape).extend(legacyQueryParamsSchema.shape).partial()
  ),
  responseSchema: listTracesResponseSchema,
  summary: "List traces",
  description: "Returns a paginated list of traces with optional filtering and sorting",
  tags: ["Observability"],
  requiresAuth: true,
  handler: async ({ mastra, ...params }) => {
    try {
      const transformedParams = transformLegacyParams(params);
      const filters = pickParams(tracesFilterSchema, transformedParams);
      const pagination = pickParams(paginationArgsSchema, transformedParams);
      const orderBy = pickParams(tracesOrderBySchema, transformedParams);
      const observabilityStore = await getObservabilityStore(mastra);
      return await observabilityStore.listTraces({ filters, pagination, orderBy });
    } catch (error) {
      return handleError(error, "Error listing traces");
    }
  }
});
var GET_TRACE_ROUTE = createRoute({
  method: "GET",
  path: "/observability/traces/:traceId",
  responseType: "json",
  pathParamSchema: getTraceArgsSchema,
  responseSchema: getTraceResponseSchema,
  summary: "Get AI trace by ID",
  description: "Returns a complete AI trace with all spans by trace ID",
  tags: ["Observability"],
  requiresAuth: true,
  handler: async ({ mastra, traceId }) => {
    try {
      const observabilityStore = await getObservabilityStore(mastra);
      const trace = await observabilityStore.getTrace({ traceId });
      if (!trace) {
        throw new HTTPException(404, { message: `Trace with ID '${traceId}' not found` });
      }
      return trace;
    } catch (error) {
      return handleError(error, "Error getting trace");
    }
  }
});
var GET_TRACE_TRAJECTORY_ROUTE = createRoute({
  method: "GET",
  path: "/observability/traces/:traceId/trajectory",
  responseType: "json",
  pathParamSchema: getTraceArgsSchema,
  responseSchema: z.object({
    steps: z.array(z.unknown()),
    totalDurationMs: z.number().optional(),
    rawOutput: z.unknown().optional(),
    rawWorkflowResult: z.unknown().optional()
  }),
  summary: "Extract trajectory from trace",
  description: "Extracts a structured trajectory (ordered steps) from a trace by analyzing its spans",
  tags: ["Observability"],
  requiresAuth: true,
  handler: async ({ mastra, traceId }) => {
    try {
      const observabilityStore = await getObservabilityStore(mastra);
      const trace = await observabilityStore.getTrace({ traceId });
      if (!trace) {
        throw new HTTPException(404, { message: `Trace with ID '${traceId}' not found` });
      }
      const trajectory = extractTrajectoryFromTrace(trace.spans);
      return trajectory;
    } catch (error) {
      return handleError(error, "Error extracting trajectory from trace");
    }
  }
});
var SCORE_TRACES_ROUTE = createRoute({
  method: "POST",
  path: "/observability/traces/score",
  responseType: "json",
  bodySchema: scoreTracesRequestSchema,
  responseSchema: scoreTracesResponseSchema,
  summary: "Score traces",
  description: "Scores one or more traces using a specified scorer (fire-and-forget)",
  tags: ["Observability"],
  requiresAuth: true,
  handler: async ({ mastra, ...params }) => {
    try {
      getStorage(mastra);
      const { scorerName, targets } = params;
      const scorer = mastra.getScorerById(scorerName);
      if (!scorer) {
        throw new HTTPException(404, { message: `Scorer '${scorerName}' not found` });
      }
      scoreTraces({
        scorerId: scorer.config.id || scorer.config.name,
        targets,
        mastra
      }).catch((error) => {
        const logger = mastra.getLogger();
        logger?.error(`Background trace scoring failed: ${error.message}`, error);
      });
      return {
        status: "success",
        message: `Scoring started for ${targets.length} ${targets.length === 1 ? "trace" : "traces"}`,
        traceCount: targets.length
      };
    } catch (error) {
      return handleError(error, "Error processing trace scoring");
    }
  }
});
var LIST_SCORES_BY_SPAN_ROUTE = createRoute({
  method: "GET",
  path: "/observability/traces/:traceId/:spanId/scores",
  responseType: "json",
  pathParamSchema: spanIdsSchema,
  // List endpoints accept optional query params; use partial() to allow empty queries.
  queryParamSchema: wrapSchemaForQueryParams(paginationArgsSchema.partial()),
  responseSchema: listScoresResponseSchema,
  summary: "List scores by span",
  description: "Returns all scores for a specific span within a trace",
  tags: ["Observability"],
  requiresAuth: true,
  handler: async ({ mastra, ...params }) => {
    try {
      const pagination = pickParams(paginationArgsSchema, params);
      const spanIds = pickParams(spanIdsSchema, params);
      const scoresStore = await getScoresStore(mastra);
      return await scoresStore.listScoresBySpan({
        ...spanIds,
        pagination
      });
    } catch (error) {
      return handleError(error, "Error getting scores by span");
    }
  }
});

export { GET_TRACE_ROUTE, GET_TRACE_TRAJECTORY_ROUTE, LIST_SCORES_BY_SPAN_ROUTE, LIST_TRACES_ROUTE, SCORE_TRACES_ROUTE, observability_exports };
