// @mastra/server/dist/chunk-DCWP744F.cjs
'use strict';

var chunkEES2ZZGL_cjs = require('./chunk-EES2ZZGL.cjs');
var v4 = require('zod/v4');

var mcpServerIdPathParams = v4.z.object({
  serverId: v4.z.string().describe("MCP server ID")
});
var mcpServerDetailPathParams = v4.z.object({
  id: v4.z.string().describe("MCP server ID")
});
var mcpServerToolPathParams = v4.z.object({
  serverId: v4.z.string().describe("MCP server ID"),
  toolId: v4.z.string().describe("Tool ID")
});
var executeToolBodySchema = v4.z.object({
  data: v4.z.unknown().optional()
});
var listMcpServersQuerySchema = chunkEES2ZZGL_cjs.createCombinedPaginationSchema();
var getMcpServerDetailQuerySchema = v4.z.object({
  version: v4.z.string().optional()
});
var versionDetailSchema = v4.z.object({
  version: v4.z.string(),
  release_date: v4.z.string(),
  is_latest: v4.z.boolean()
});
var serverInfoSchema = v4.z.object({
  id: v4.z.string(),
  name: v4.z.string(),
  version_detail: versionDetailSchema
});
var listMcpServersResponseSchema = v4.z.object({
  servers: v4.z.array(serverInfoSchema),
  total_count: v4.z.number(),
  next: v4.z.string().nullable()
});
var serverDetailSchema = v4.z.object({
  id: v4.z.string(),
  name: v4.z.string(),
  description: v4.z.string().optional(),
  version_detail: versionDetailSchema,
  package_canonical: v4.z.string().optional(),
  packages: v4.z.array(v4.z.unknown()).optional(),
  remotes: v4.z.array(v4.z.unknown()).optional()
});
var mcpToolInfoSchema = v4.z.object({
  name: v4.z.string(),
  description: v4.z.string().optional(),
  inputSchema: v4.z.unknown(),
  outputSchema: v4.z.unknown().optional(),
  toolType: v4.z.string().optional()
});
var listMcpServerToolsResponseSchema = v4.z.object({
  tools: v4.z.array(mcpToolInfoSchema)
});
var executeToolResponseSchema = v4.z.object({
  result: v4.z.unknown()
});
var jsonRpcErrorSchema = v4.z.object({
  jsonrpc: v4.z.literal("2.0"),
  error: v4.z.object({
    code: v4.z.number(),
    message: v4.z.string()
  }),
  id: v4.z.null()
});

exports.executeToolBodySchema = executeToolBodySchema;
exports.executeToolResponseSchema = executeToolResponseSchema;
exports.getMcpServerDetailQuerySchema = getMcpServerDetailQuerySchema;
exports.jsonRpcErrorSchema = jsonRpcErrorSchema;
exports.listMcpServerToolsResponseSchema = listMcpServerToolsResponseSchema;
exports.listMcpServersQuerySchema = listMcpServersQuerySchema;
exports.listMcpServersResponseSchema = listMcpServersResponseSchema;
exports.mcpServerDetailPathParams = mcpServerDetailPathParams;
exports.mcpServerIdPathParams = mcpServerIdPathParams;
exports.mcpServerToolPathParams = mcpServerToolPathParams;
exports.mcpToolInfoSchema = mcpToolInfoSchema;
exports.serverDetailSchema = serverDetailSchema;
exports.serverInfoSchema = serverInfoSchema;
exports.versionDetailSchema = versionDetailSchema;
