// @mastra/server/dist/chunk-NIKW5NHG.js
import { z } from 'zod/v4';

// src/server/schemas/tool-providers.ts
var toolProviderIdPathParams = z.object({
  providerId: z.string().describe("Unique identifier for the tool provider")
});
var toolSlugPathParams = toolProviderIdPathParams.extend({
  toolSlug: z.string().describe("Slug identifier for the tool")
});
var listToolProviderToolsQuerySchema = z.object({
  toolkit: z.string().optional().describe("Filter tools by toolkit slug"),
  search: z.string().optional().describe("Search tools by name or description"),
  page: z.coerce.number().optional().describe("Page number for pagination"),
  perPage: z.coerce.number().optional().describe("Number of items per page")
});
var paginationSchema = z.object({
  total: z.number().optional(),
  page: z.number().optional(),
  perPage: z.number().optional(),
  hasMore: z.boolean()
}).optional();
var listToolProvidersResponseSchema = z.object({
  providers: z.array(
    z.object({
      id: z.string(),
      name: z.string(),
      description: z.string().optional()
    })
  )
});
var listToolProviderToolkitsResponseSchema = z.object({
  data: z.array(
    z.object({
      slug: z.string(),
      name: z.string(),
      description: z.string().optional(),
      icon: z.string().optional()
    })
  ),
  pagination: paginationSchema
});
var listToolProviderToolsResponseSchema = z.object({
  data: z.array(
    z.object({
      slug: z.string(),
      name: z.string(),
      description: z.string().optional(),
      toolkit: z.string().optional()
    })
  ),
  pagination: paginationSchema
});
var getToolProviderToolSchemaResponseSchema = z.record(z.string(), z.unknown());

export { getToolProviderToolSchemaResponseSchema, listToolProviderToolkitsResponseSchema, listToolProviderToolsQuerySchema, listToolProviderToolsResponseSchema, listToolProvidersResponseSchema, toolProviderIdPathParams, toolSlugPathParams };
