// @mastra/server/dist/chunk-AJYJ4H7P.js
import { randomUUID } from 'crypto';
import { isProviderDefinedTool } from '@mastra/core/tools';
import { zodToJsonSchema } from '@mastra/core/utils/zod-to-json';

// src/server/handlers/responses.adapter.ts
function normalizeMessageContent(content) {
  if (typeof content === "string") {
    return content;
  }
  return content.map((part) => part.text).join("");
}
function getMessageText(message) {
  const parts = Array.isArray(message.content?.parts) ? message.content.parts : [];
  return parts.flatMap((part) => part.type === "text" ? [part.text] : []).filter((text) => typeof text === "string").join("");
}
function isRecord(value) {
  return typeof value === "object" && value !== null && !Array.isArray(value);
}
function getMessageRole(message) {
  return message.role ?? "";
}
function getToolKey(toolCallId, messageId, partIndex) {
  return `${messageId}:${partIndex}`;
}
function normalizeToolParameters(schema) {
  if (!isRecord(schema)) {
    return schema;
  }
  if (isRecord(schema.json) && Object.keys(schema).length === 1) {
    return schema.json;
  }
  return schema;
}
function mapMastraToolsToResponseTools(tools) {
  if (!tools) {
    return [];
  }
  return Object.values(tools).flatMap((tool) => {
    if (!isRecord(tool)) {
      return [];
    }
    const name = typeof tool.id === "string" ? tool.id : typeof tool.name === "string" ? tool.name : null;
    if (!name) {
      return [];
    }
    const description = typeof tool.description === "string" ? tool.description : void 0;
    let parameters;
    if (isProviderDefinedTool(tool)) {
      const resolvedSchema = typeof tool.inputSchema === "function" ? tool.inputSchema() : tool.inputSchema;
      parameters = isRecord(resolvedSchema) && "jsonSchema" in resolvedSchema ? normalizeToolParameters(resolvedSchema.jsonSchema) : void 0;
    } else if ("inputSchema" in tool && tool.inputSchema) {
      parameters = normalizeToolParameters(zodToJsonSchema(tool.inputSchema));
    }
    return [
      {
        type: "function",
        name,
        ...description ? { description } : {},
        ...parameters !== void 0 ? { parameters: JSON.parse(JSON.stringify(parameters)) } : {}
      }
    ];
  });
}
function stringifyToolPayload(value) {
  if (typeof value === "string") {
    return value;
  }
  return JSON.stringify(value ?? {});
}
function createOutputMessage({
  messageId,
  status,
  text
}) {
  const responseStatus = status === "completed" ? "completed" : "incomplete";
  return {
    id: messageId,
    type: "message",
    role: "assistant",
    status: responseStatus,
    content: [createOutputTextPart(text)]
  };
}
function createConversationMessage({
  messageId,
  role,
  text
}) {
  return {
    id: messageId,
    type: "message",
    role,
    status: "completed",
    content: [
      role === "assistant" ? createOutputTextPart(text) : {
        type: "input_text",
        text
      }
    ]
  };
}
function createFunctionCallItem({
  itemId,
  callId,
  name,
  args
}) {
  return {
    id: itemId,
    type: "function_call",
    call_id: callId,
    name,
    arguments: stringifyToolPayload(args),
    status: "completed"
  };
}
function createFunctionCallOutputItem({ itemId, callId, output }) {
  return {
    id: itemId,
    type: "function_call_output",
    call_id: callId,
    output: stringifyToolPayload(output)
  };
}
function collectToolResultCallIds(messages) {
  const toolResultCallIds = /* @__PURE__ */ new Set();
  for (const message of messages) {
    const parts = Array.isArray(message.content?.parts) ? message.content.parts : [];
    for (const [partIndex, part] of parts.entries()) {
      if (!isRecord(part) || part.type !== "tool-invocation" || !isRecord(part.toolInvocation)) {
        continue;
      }
      const toolInvocation = part.toolInvocation;
      const toolCallId = typeof toolInvocation.toolCallId === "string" ? toolInvocation.toolCallId : getToolKey(null, message.id, partIndex);
      if (getMessageRole(message) === "tool" && toolInvocation.result !== void 0) {
        toolResultCallIds.add(toolCallId);
      }
    }
  }
  return toolResultCallIds;
}
function mapMastraMessageToResponseToolItems({
  message,
  toolResultCallIds,
  emittedCallIds,
  emittedResultCallIds
}) {
  const items = [];
  const parts = Array.isArray(message.content?.parts) ? message.content.parts : [];
  for (const [partIndex, part] of parts.entries()) {
    if (!isRecord(part) || part.type !== "tool-invocation" || !isRecord(part.toolInvocation)) {
      continue;
    }
    const toolInvocation = part.toolInvocation;
    const toolName = typeof toolInvocation.toolName === "string" ? toolInvocation.toolName : null;
    const toolCallId = typeof toolInvocation.toolCallId === "string" ? toolInvocation.toolCallId : getToolKey(null, message.id, partIndex);
    if (getMessageRole(message) === "assistant" && toolName && !emittedCallIds.has(toolCallId)) {
      items.push(
        createFunctionCallItem({
          itemId: `${message.id}:${partIndex}:call`,
          callId: toolCallId,
          name: toolName,
          args: toolInvocation.args
        })
      );
      emittedCallIds.add(toolCallId);
    }
    if (toolInvocation.result !== void 0 && !emittedResultCallIds.has(toolCallId) && (getMessageRole(message) === "tool" || !toolResultCallIds.has(toolCallId))) {
      items.push(
        createFunctionCallOutputItem({
          itemId: `${message.id}:${partIndex}:output`,
          callId: toolCallId,
          output: toolInvocation.result
        })
      );
      emittedResultCallIds.add(toolCallId);
    }
  }
  return items;
}
function mapMastraMessagesToConversationItems(messages) {
  if (!messages.length) {
    return [];
  }
  const items = [];
  const toolResultCallIds = collectToolResultCallIds(messages);
  const emittedCallIds = /* @__PURE__ */ new Set();
  const emittedResultCallIds = /* @__PURE__ */ new Set();
  for (const message of messages) {
    items.push(
      ...mapMastraMessageToResponseToolItems({
        message,
        toolResultCallIds,
        emittedCallIds,
        emittedResultCallIds
      })
    );
    const role = getMessageRole(message);
    const text = getMessageText(message);
    if ((role === "user" || role === "system" || role === "assistant") && text) {
      items.push(
        createConversationMessage({
          messageId: message.id,
          role,
          text
        })
      );
      continue;
    }
    if (role === "assistant" && !text) {
      const parts = Array.isArray(message.content?.parts) ? message.content.parts : [];
      const hasOnlyToolInvocations = parts.every(
        (part) => isRecord(part) && part.type === "tool-invocation" && isRecord(part.toolInvocation)
      );
      if (hasOnlyToolInvocations) {
        continue;
      }
    }
    if (role === "tool") {
      continue;
    }
  }
  return items;
}
function mapMastraMessagesToResponseOutputItems({
  messages,
  outputMessageId,
  status,
  fallbackText
}) {
  if (!messages?.length) {
    return [createOutputMessage({ messageId: outputMessageId, status, text: fallbackText })];
  }
  const output = [];
  const lastAssistantIndex = [...messages].map((message) => message.role).lastIndexOf("assistant");
  const toolResultCallIds = collectToolResultCallIds(messages);
  const emittedCallIds = /* @__PURE__ */ new Set();
  const emittedResultCallIds = /* @__PURE__ */ new Set();
  for (const [messageIndex, message] of messages.entries()) {
    output.push(
      ...mapMastraMessageToResponseToolItems({
        message,
        toolResultCallIds,
        emittedCallIds,
        emittedResultCallIds
      })
    );
    const text = getMessageText(message);
    if (getMessageRole(message) === "assistant" && text) {
      output.push(
        createOutputMessage({
          messageId: messageIndex === lastAssistantIndex ? outputMessageId : message.id,
          status,
          text
        })
      );
    }
  }
  if (!output.some((item) => item.type === "message") && fallbackText) {
    output.push(createOutputMessage({ messageId: outputMessageId, status, text: fallbackText }));
  }
  return output;
}
function createMessageId() {
  return `msg_${randomUUID()}`;
}
function mapResponseInputToExecutionMessages(input) {
  if (typeof input === "string") {
    return [{ role: "user", content: input }];
  }
  return input.map((message) => ({
    role: message.role === "developer" ? "system" : message.role,
    content: normalizeMessageContent(message.content)
  }));
}
function toResponseUsage(usage) {
  if (!usage) {
    return null;
  }
  const inputTokens = usage.inputTokens ?? usage.promptTokens ?? 0;
  const outputTokens = usage.outputTokens ?? usage.completionTokens ?? 0;
  const totalTokens = usage.totalTokens ?? inputTokens + outputTokens;
  return {
    input_tokens: inputTokens,
    output_tokens: outputTokens,
    total_tokens: totalTokens,
    input_tokens_details: {
      cached_tokens: 0
    },
    output_tokens_details: {
      reasoning_tokens: 0
    }
  };
}
function toResponseStatus(finishReason) {
  if (finishReason === "suspended" || finishReason === "error") {
    return "incomplete";
  }
  return "completed";
}
function createOutputTextPart(text) {
  return {
    type: "output_text",
    text,
    annotations: [],
    logprobs: []
  };
}
function buildCompletedResponse({
  responseId,
  outputMessageId,
  model,
  createdAt,
  completedAt,
  status,
  text,
  usage,
  instructions,
  textConfig,
  previousResponseId,
  conversationId,
  providerOptions,
  tools,
  store,
  messages
}) {
  return {
    id: responseId,
    object: "response",
    created_at: createdAt,
    completed_at: completedAt,
    model,
    status,
    output: mapMastraMessagesToResponseOutputItems({
      messages,
      outputMessageId,
      status,
      fallbackText: text
    }),
    usage: toResponseUsage(usage),
    error: null,
    incomplete_details: null,
    instructions: instructions ?? null,
    text: textConfig ?? null,
    previous_response_id: previousResponseId ?? null,
    conversation_id: conversationId ?? null,
    providerOptions,
    tools,
    store
  };
}
function buildInProgressResponse({
  responseId,
  model,
  createdAt,
  instructions,
  textConfig,
  previousResponseId,
  conversationId,
  tools,
  store
}) {
  return {
    id: responseId,
    object: "response",
    created_at: createdAt,
    completed_at: null,
    model,
    status: "in_progress",
    output: [],
    usage: null,
    error: null,
    incomplete_details: null,
    instructions: instructions ?? null,
    text: textConfig ?? null,
    previous_response_id: previousResponseId ?? null,
    conversation_id: conversationId ?? null,
    tools: tools ?? [],
    store
  };
}
function mapResponseTurnRecordToResponse(match) {
  return {
    id: match.message.id,
    object: "response",
    created_at: match.metadata.createdAt,
    completed_at: match.metadata.completedAt,
    model: match.metadata.model,
    status: match.metadata.status,
    output: mapMastraMessagesToResponseOutputItems({
      messages: match.messages,
      outputMessageId: match.message.id,
      status: match.metadata.status,
      fallbackText: getMessageText(match.message)
    }),
    usage: match.metadata.usage,
    error: null,
    incomplete_details: null,
    instructions: match.metadata.instructions ?? null,
    text: match.metadata.text ?? null,
    previous_response_id: match.metadata.previousResponseId ?? null,
    conversation_id: match.thread.id,
    providerOptions: match.metadata.providerOptions,
    tools: match.metadata.tools,
    store: match.metadata.store
  };
}
function formatSseEvent(event, data) {
  return new TextEncoder().encode(`event: ${event}
data: ${JSON.stringify(data)}

`);
}
function extractTextDelta(value) {
  if (typeof value !== "object" || value === null || !("type" in value)) {
    return null;
  }
  const chunk = value;
  switch (chunk.type) {
    case "text-delta":
      if (typeof chunk.payload?.text === "string") {
        return chunk.payload.text;
      }
      if (typeof chunk.textDelta === "string") {
        return chunk.textDelta;
      }
      if (typeof chunk.text === "string") {
        return chunk.text;
      }
      return null;
    default:
      return null;
  }
}

export { buildCompletedResponse, buildInProgressResponse, createMessageId, createOutputTextPart, extractTextDelta, formatSseEvent, mapMastraMessagesToConversationItems, mapMastraMessagesToResponseOutputItems, mapMastraToolsToResponseTools, mapResponseInputToExecutionMessages, mapResponseTurnRecordToResponse, toResponseStatus, toResponseUsage };
