// @mastra/server/dist/chunk-DOHUOYZS.cjs
'use strict';

var chunkHITLRKIU_cjs = require('./chunk-HITLRKIU.cjs');
var v4 = require('zod/v4');

var actionIdPathParams = v4.z.object({
  actionId: v4.z.string().describe("Unique identifier for the agent-builder action")
});
var actionRunPathParams = v4.z.object({
  actionId: v4.z.string().describe("Unique identifier for the agent-builder action"),
  runId: v4.z.string().describe("Unique identifier for the action run")
});
var streamAgentBuilderBodySchema = chunkHITLRKIU_cjs.streamWorkflowBodySchema;
var streamLegacyAgentBuilderBodySchema = chunkHITLRKIU_cjs.streamLegacyWorkflowBodySchema;
var resumeAgentBuilderBodySchema = chunkHITLRKIU_cjs.resumeBodySchema;
var startAsyncAgentBuilderBodySchema = chunkHITLRKIU_cjs.startAsyncWorkflowBodySchema;

exports.actionIdPathParams = actionIdPathParams;
exports.actionRunPathParams = actionRunPathParams;
exports.resumeAgentBuilderBodySchema = resumeAgentBuilderBodySchema;
exports.startAsyncAgentBuilderBodySchema = startAsyncAgentBuilderBodySchema;
exports.streamAgentBuilderBodySchema = streamAgentBuilderBodySchema;
exports.streamLegacyAgentBuilderBodySchema = streamLegacyAgentBuilderBodySchema;
